---
title: "Data QC"
author: "Heath O'Brien"
output:
  tufte::tufte_html: default
  #tufte::tufte_handout: default
---
<!-- see http://rstudio.github.io/tufte/ for info about tufte -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(tidyr)
library(ggplot2)
library(dplyr)
library(gridExtra)
library(RColorBrewer)
library(tufte)
rm(list=ls())
source("~/LabNotes/R/FormatGGplot.R")
folders <- c("/c8000xd3/rnaseq-heath/Mapping/15533/15533",
             "/c8000xd3/rnaseq-heath/Mapping/16438/16438",
             "/c8000xd3/rnaseq-heath/Mapping/16972/16972",
             "/c8000xd3/rnaseq-heath/Mapping/17087/17087",
             "/c8000xd3/rnaseq-heath/Mapping/17109/17109",
             "/c8000xd3/rnaseq-heath/Mapping/17701/17701",
             "/c8000xd3/rnaseq-heath/Mapping/18294/18294",
             "/c8000xd3/rnaseq-heath/Mapping/18655/18655",
             "/c8000xd3/rnaseq-heath/Mapping/15768/15768",
             "/c8000xd3/rnaseq-heath/Mapping/16286/16286",
             "/c8000xd3/rnaseq-heath/Mapping/17054/17054",
             "/c8000xd3/rnaseq-heath/Mapping/17068/17068",
             "/c8000xd3/rnaseq-heath/Mapping/17081/17081",
             "/c8000xd3/rnaseq-heath/Mapping/17115/17115",
             "/c8000xd3/rnaseq-heath/Mapping/17130/17130",
             "/c8000xd3/rnaseq-heath/Mapping/17812/17812",
             "/c8000xd3/rnaseq-heath/Mapping/18349/18349",
             "/c8000xd3/rnaseq-heath/Mapping/18596/18596",
             "/c8000xd3/rnaseq-heath/Mapping/15468/15468",
             "/c8000xd3/rnaseq-heath/Mapping/16488/16488",
             "/c8000xd3/rnaseq-heath/Mapping/16840/16840",
             "/c8000xd3/rnaseq-heath/Mapping/16929/16929",
             "/c8000xd3/rnaseq-heath/Mapping/17049/17049",
             "/c8000xd3/rnaseq-heath/Mapping/18687/18687",
             "/c8000xd3/rnaseq-heath/Mapping/15641/15641",
             "/c8000xd3/rnaseq-heath/Mapping/16024/16024",
             "/c8000xd3/rnaseq-heath/Mapping/16115/16115",
             "/c8000xd3/rnaseq-heath/Mapping/16385/16385",
             "/c8000xd3/rnaseq-heath/Mapping/16428/16428",
             "/c8000xd3/rnaseq-heath/Mapping/16491/16491",
             "/c8000xd3/rnaseq-heath/Mapping/16548/16548",
             "/c8000xd3/rnaseq-heath/Mapping/16810/16810",
             "/c8000xd3/rnaseq-heath/Mapping/16826/16826",
             "/c8000xd3/rnaseq-heath/Mapping/17048/17048",
             "/c8000xd3/rnaseq-heath/Mapping/17053/17053",
             "/c8000xd3/rnaseq-heath/Mapping/17071/17071",
             "/c8000xd3/rnaseq-heath/Mapping/17921-l1/17921-l1",
             "/c8000xd3/rnaseq-heath/Mapping/15655/15655",
             "/c8000xd3/rnaseq-heath/Mapping/16483/16483",
             "/c8000xd3/rnaseq-heath/Mapping/16640/16640",
             "/c8000xd3/rnaseq-heath/Mapping/17013/17013",
             # < 2000 reads "/c8000xd3/rnaseq-heath/Mapping/17025/17025",
             "/c8000xd3/rnaseq-heath/Mapping/17198/17198",
             "/c8000xd3/rnaseq-heath/Mapping/17229/17229",
             "/c8000xd3/rnaseq-heath/Mapping/17333/17333",
             "/c8000xd3/rnaseq-heath/Mapping/17475/17475",
             "/c8000xd3/rnaseq-heath/Mapping/17543/17543",
             "/c8000xd3/rnaseq-heath/Mapping/17629/17629",
             "/c8000xd3/rnaseq-heath/Mapping/17835/17835",
             "/c8000xd3/rnaseq-heath/Mapping/18249/18249", 
             "/c8000xd3/rnaseq-heath/Mapping/18372/18372",
             "/c8000xd3/rnaseq-heath/Mapping/18666/18666",
             "/c8000xd3/rnaseq-heath/Mapping/18983/18983",
             "/c8000xd3/rnaseq-heath/Mapping/19043/19043",
             "/c8000xd3/rnaseq-heath/Mapping/19052/19052",
             "/c8000xd3/rnaseq-heath/Mapping/A138/A138",
             "/c8000xd3/rnaseq-heath/Mapping/A226/A226",
             "/c8000xd3/rnaseq-heath/Mapping/15240/15240",
             "/c8000xd3/rnaseq-heath/Mapping/16649/16649",
             "/c8000xd3/rnaseq-heath/Mapping/17072/17072",
             "/c8000xd3/rnaseq-heath/Mapping/17160/17160",
             "/c8000xd3/rnaseq-heath/Mapping/17167/17167",
             "/c8000xd3/rnaseq-heath/Mapping/17175/17175",
             "/c8000xd3/rnaseq-heath/Mapping/17671/17671",
             "/c8000xd3/rnaseq-heath/Mapping/17922/17922",
             "/c8000xd3/rnaseq-heath/Mapping/17923/17923",
             "/c8000xd3/rnaseq-heath/Mapping/18121/18121",
             "/c8000xd3/rnaseq-heath/Mapping/18134/18134",
             "/c8000xd3/rnaseq-heath/Mapping/18153/18153",
             "/c8000xd3/rnaseq-heath/Mapping/18241/18241",
             "/c8000xd3/rnaseq-heath/Mapping/18266/18266",
             "/c8000xd3/rnaseq-heath/Mapping/18355/18355",
             "/c8000xd3/rnaseq-heath/Mapping/18432/18432",
             "/c8000xd3/rnaseq-heath/Mapping/18559/18559",
             "/c8000xd3/rnaseq-heath/Mapping/18694/18694",
             "/c8000xd3/rnaseq-heath/Mapping/17369/17369",
             "/c8000xd3/rnaseq-heath/Mapping/18055/18055",
             "/c8000xd3/rnaseq-heath/Mapping/15533-2/15533-2",
             "/c8000xd3/rnaseq-heath/Mapping/18282/18282",
             "/c8000xd3/rnaseq-heath/Mapping/15329/15329",
             "/c8000xd3/rnaseq-heath/Mapping/16117/16117",
             "/c8000xd3/rnaseq-heath/Mapping/16859/16859",
             "/c8000xd3/rnaseq-heath/Mapping/17221/17221",
             "/c8000xd3/rnaseq-heath/Mapping/17372/17372",
             "/c8000xd3/rnaseq-heath/Mapping/18653/18653",
             "/c8000xd3/rnaseq-heath/Mapping/19031/19031",
             "/c8000xd3/rnaseq-heath/Mapping/A138-2/A138-2",
             "/c8000xd3/rnaseq-heath/Mapping/A151/A151",
             "/c8000xd3/rnaseq-heath/Mapping/A226-2/A226-2",
             "/c8000xd3/rnaseq-heath/Mapping/A254/A254",
             "/c8000xd3/rnaseq-heath/Mapping/A263/A263",
             "/c8000xd3/rnaseq-heath/Mapping/17753/17753",
             "/c8000xd3/rnaseq-heath/Mapping/18139/18139",
             "/c8000xd3/rnaseq-heath/Mapping/A139/A139",
             "/c8000xd3/rnaseq-heath/Mapping/17932/17932",
             "/c8000xd3/rnaseq-heath/Mapping/18153-2/18153-2",
             "/c8000xd3/rnaseq-heath/Mapping/17486/17486",
             "/c8000xd3/rnaseq-heath/Mapping/16483-2/16483-2",
             "/c8000xd3/rnaseq-heath/Mapping/17162/17162",
             "/c8000xd3/rnaseq-heath/Mapping/18528/18528",
             "/c8000xd3/rnaseq-heath/Mapping/A145/A145",
             "/c8000xd3/rnaseq-heath/Mapping/17521/17521",
             "/c8000xd3/rnaseq-heath/Mapping/17754/17754",
             "/c8000xd3/rnaseq-heath/Mapping/18856/18856",
             "/c8000xd3/rnaseq-heath/Mapping/18015/18015",
             "/c8000xd3/rnaseq-heath/Mapping/17054-2/17054-2",
             "/c8000xd3/rnaseq-heath/Mapping/18611/18611",
             "/c8000xd3/rnaseq-heath/Mapping/18382/18382",
             "/c8000xd3/rnaseq-heath/Mapping/18529/18529",
             "/c8000xd3/rnaseq-heath/Mapping/17264/17264",
             "/c8000xd3/rnaseq-heath/Mapping/17111/17111",
             "/c8000xd3/rnaseq-heath/Mapping/18655-2/18655-2",
             "/c8000xd3/rnaseq-heath/Mapping/17666/17666",
             "/c8000xd3/rnaseq-heath/Mapping/17046/17046",
             "/c8000xd3/rnaseq-heath/Mapping/17193/17193",
             "/c8000xd3/rnaseq-heath/Mapping/18540/18540",
             "/c8000xd3/rnaseq-heath/Mapping/15296/15296"             )
LibraryInfo <- read.delim("/Users/heo3/BTSync/FetalRNAseq/LabNotes/SampleProgress.txt", 
                     header=TRUE, stringsAsFactors=TRUE, sep='\t'
                     )
```

# RSeQC
## Mapping Stats
`r margin_note("- rDNA reads not included in first three stats")`
`r margin_note("- # uniquely mapped reads ranges from 57 million to 446 million, with 3 samples (17054, 17160, 15483) having < 100 million")`
`r margin_note("- % rRNA ranges from 0.1% to 56%, with 4 samples (17054, 17160, 18655, 16428) having > 40%")`

``` {r }
#bam_stat.py
RSeQCstats <- data.frame()
for (folder in folders) {
  #print(folder)
  if(file.exists(paste(folder, ".ex.stats.txt", sep=""))){
    temp <- read.delim(paste(folder, ".ex.stats.txt", sep=""), 
                header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':'
                )
    } else {
    temp <- read.delim(paste(folder, ".chr.stats.txt", sep=""), 
                         header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':'
                         )
    }
  temp[4,2] <-strsplit(temp[4,1], ' +')[[1]][4]
  temp[4,1] <- 'Non primary hits'
  temp <- temp[c(6,7,14),]
  temp2 <- read.delim(paste(folder, ".in.stats.txt", sep=""), 
                     header=FALSE, stringsAsFactors=FALSE, skip=5, sep=':'
                     )
  temp2[4,2] <-strsplit(temp[4,1], ' +')[[1]][4]
  temp2[4,1] <- 'Non primary hits'
  temp <-rbind(temp, c("rDNA",sum(as.numeric(temp2[c(6,7),2]))))
  temp[,1] <- c("Multimapped", "Unique", "Paired", "rDNA")
  temp$sample <- tail(strsplit(folder, '/')[[1]], 1)
  RSeQCstats <- rbind(RSeQCstats, temp)
}
RSeQCstats <- spread(RSeQCstats, V1, V2)
#mutate(RSeQCstats, Perc_rDNA = as.numeric(rDNA)/(as.numeric(Unique)+as.numeric(Multimapped)+as.numeric(rDNA))) %>% arrange(Perc_rDNA)
ReadNumbers <- RSeQCstats[,c(1,5)]
RSeQCstats <- RSeQCstats[c(1,5,3,2,4)]
RSeQCstats[,2:5] <- sapply(RSeQCstats[,2:5], function(x) formatC(as.numeric(x), format='d', big.mark=','))
knitr::kable(
  RSeQCstats
)
```


## Distribution of Reads Among Features (proportion of tags mapping to different feature classes)
`r margin_note("- Intergenic only considers sequence within 10 kb upstream of TSS and downstream of TES")`
`r margin_note("- Exons enriched >10 fold relative to introns")`
`r margin_note("- Introns enriched >10 fold relative to Intergenic")`
`r margin_note("- 1/3rd to 2/3rd of tags map to introns. 2-4% map to intergenic (except for 17198 which is 13% intergenic)")`
`r margin_note("- Number of tags per read ranges from 1.1 to 1.8, except for 17198, which has 0.8")`
`r margin_note("- 17198, also has the highest proportion of intronic and intergenic sequence and the lowest proportion of CDS and second lowest of UTR")`
`r margin_note("- I wonder if this might be an indeciation of DNA contamination?")`


``` {r }
#read_distribution.py
RSeQCdistribution <- data.frame()
ReadNumbers$ExpressedTags = NA
for (folder in folders) {
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp <- read.delim(paste(folder, ".dist.txt", sep=""), 
                           header=TRUE, stringsAsFactors=FALSE, skip=4, sep=''
                     )
  temp <- temp[-11,]
  tag_total <- sum(temp[c(1,2,3,4,7,10),3])
  ReadNumbers[ReadNumbers$sample == sample,]$ExpressedTags <- sum(temp[c(1,2,3),3])
  temp <- data.frame(V1 = c("Tags_per_read", "CDS", "UTR", "Intron", "Intergenic"), 
                     V2=c(
                  tag_total/as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$Unique),
                  temp[1,3]/tag_total, 
                  sum(as.numeric(temp[c(2,3),3]))/tag_total,
                  temp[4,3]/tag_total,
                  sum(as.numeric(temp[c(7,10),3]))/tag_total
                  )
             )
  temp$sample <- sample
  #temp$centre <- LibraryInfo[LibraryInfo$sample == sample,2]
  RSeQCdistribution <- rbind(RSeQCdistribution, temp)
}
RSeQCdistribution <- spread(RSeQCdistribution, V1, V2)
RSeQCdistribution <- RSeQCdistribution[c(1,5,2,6,4,3)]

knitr::kable(
  RSeQCdistribution
)
```

## Strand

`r margin_note("- Data consistent with second-strand experiment")`

```{r }
#infer_experiment.py
RSeQCstats <- data.frame()
for (folder in folders) {
  temp <- read.delim(paste(folder, ".expt.txt", sep=""), 
                     skip=3, header=FALSE, sep=':'
                     )
  temp$sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$V1 <- c("Ambiguous", "First Strand", "Second Strand")
  RSeQCstats <- rbind(RSeQCstats, temp)
}
RSeQCstats <- spread(RSeQCstats, V1, V2)

knitr::kable(
  RSeQCstats
)
```

## Insert Size Distribution

`r margin_note("- Sharp peaks indicate that most fragments are shorter than the read length")`
`r margin_note("- Inner distance pretty consistenly at - 125 for Edinbrugh data, but centered at about -50 for Exeter data")`
`r margin_note("- This indicates that fragment size tends to be around 125-150 bp")`


```{r include=FALSE}
#inner_distance.py
RSeQCstats <- data.frame()
for (folder in folders) {
  #print(folder)
  temp <- read.delim(paste(folder, ".inner_distance_freq.txt", sep=""),
                     header=FALSE
                     )
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$sample <- sample
  temp$centre <- LibraryInfo[LibraryInfo$sample == sample,2]
  RSeQCstats <- rbind(RSeQCstats, temp)
}
```

```{r dev.args = list(bg = 'transparent')}
ggplot(RSeQCstats, aes(x=(V1+V2)/2, y=V3, colour=centre, group=sample)) +
  geom_line() +
  ylab('reads') +
  xlab('inner distance') +
  theme(legend.position=c(.8,.8)) +
  scale_colour_brewer(type = "qual", palette = 6) +
  tufte_theme() 

```

## Transcript Read Coverage

`r margin_note("- Looks good. Steep drop-offs at the ends, especially 5'")`
`r margin_note("- I'm still a bit surprised about the spread in these values because I've normalised by the number of tags that map to CDS or UTR features. I guess there can be some variation in the size of the tags, but this still seems like a lot")`

```{r dev.args = list(bg = 'transparent')}
#geneBody_coverage.py
RSeQCstats <- data.frame()
for (folder in folders) {
  #print(folder)
  temp <- read.delim(paste(folder, ".geneBodyCoverage.txt", sep=""),
                     header=FALSE
                     )
  temp <-  as.data.frame(t(temp[,-1]))
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$sample <- sample
  temp$centre <- LibraryInfo[LibraryInfo$sample == sample,2]
  temp$V2 <- temp$V2
  temp$ReadNumbers <- as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$ExpressedTags)
  RSeQCstats <- rbind(RSeQCstats, temp)
}

#ggplot(subset(RSeQCstats, sample != 17175), aes(x=V1, y=V2/ReadNumbers, colour=centre, group=sample)) +
ggplot(RSeQCstats, aes(x=V1, y=V2/ReadNumbers, colour=centre, group=sample)) +
  geom_line() +
  ylab('reads') +
  tufte_theme() +
  theme(legend.position=c(0.1,.9)) +
  scale_colour_brewer(type = "qual", palette = 6) +
  xlab('transcript percentile') +
  ylab("proportion of expressed tags")
  
```

## Sequence Duplication

`r margin_note("- Plenty of duplication")`
`r margin_note("- Probably not unexpected in this case")`
`r margin_note("- Duplication higher for Exeter samples")`

```{r include=FALSE}
#read_duplication.py
RSeQCstats <- data.frame()
for (folder in folders) {
  eval(parse(file = paste(folder, ".DupRate_plot.r", sep=""))[2:6])
  temp <- rbind(data.frame(Reads=pos_uniqRead, 
                            Duplication_level=pos_occ, 
                            Method='Position-based'
                            ),
                 data.frame(Reads=seq_uniqRead, 
                            Duplication_level=seq_occ, 
                            Method='Sequence-based'
                            )
                )
  sample <- tail(strsplit(folder, '/')[[1]], 1)
  temp$sample <- sample
  temp$centre <- LibraryInfo[LibraryInfo$sample == sample,2]
  temp$ReadNumbers <- as.numeric(ReadNumbers[ReadNumbers$sample == sample,]$Unique)
  RSeQCstats <- rbind(RSeQCstats, temp)
}

```

```{r dev.args = list(bg = 'transparent'), warning=FALSE}
ggplot(RSeQCstats, aes(x=Duplication_level, y=Reads/ReadNumbers, colour=centre, group=sample)) +
  geom_point(alpha=.2) +
  xlim(c(0,500)) +
  scale_y_log10() +
  tufte_theme() +
  theme(legend.position=c(.8,.8)) +
  scale_colour_brewer(type = "qual", palette = 6) +
  facet_wrap(~ Method) +
  ylab("Proportion of Reads")


```

## Splice Junction Saturation

`r margin_note("- It looks like both known and novel splice sites continue to be found")`

