---
title: "DESeq2 Results"
author: "Heath O'Brien"
date: "`r Sys.Date()`"
output:
  tufte::tufte_html: default
---
<!-- see http://rstudio.github.io/tufte/ for info about tufte -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warnings=FALSE, message = FALSE, dev.args = list(bg = 'transparent'))
rm(list=ls())

library(readr)
library(dplyr)
library(tidyr)
library(knitr)
library(ggplot2)
library(tufte)
source("~/BTSync/FetalRNAseq/LabNotes/R/FormatGGplot.R")
```

# Running DEseq2 via SARTools (at FDR < 0.05)

`r margin_note("- All samples <14 weeks are from HDBR and all samples >20 weeks are not")`
`r margin_note("- Model for Wald tests is ~ Sex + PCW + RIN + Centre")`
`r margin_note("- LRT is ~ Sex * PCW + RIN + Centre vs. ~ PCW + RIN + Centre")`
`r margin_note("- Very few DE genes when samples > 14 weeks included, and most are on sex chromosomes")`
`r margin_note("- Excluding non HDBR samples increases the number of DE genes")`
`r margin_note("- LRT results in slightly more DE genes when analysing all HDBR samples, but far less when analysing samples > 14 PCW")`


```{r}
DESeq2results <- data.frame(BrainBank=c( "All", 
                                         "HDBR only", 
                                         rep("All", 2), 
                                         rep("HDBR only", 10)
                                         ), 
                            AgeRange=c("12-22", 
                                       "12-20", 
                                       "14-22", 
                                       rep("14-20", 2),
                                       "12-14",
                                       "12-20",
                                       rep("14-20", 7)
                                       ),
                            RIN=c(rep("All", 14)),
                            Excluded=c(rep("None", 8), 
                                       "15641", 
                                       rep("15641/18432", 4), 
                                       "15641/18432/16428"
                                       ),
                            n=c(108, 100, 61, 58, 53, 47, 100, 53, 52, rep(51, 4), 50),
                            test=c(rep("Wald", 6), rep("LRT", 2), rep("Wald", 6)),
                            CooksCutoff=c(rep("None", 10), 2, 1, 0.75, "None"),
                            DEgenes=c(68, 105, 633, 1747, 2665, 
                                      47, 154, 868, 2305, 2961, 1738, 1763, 1756, 1170),
                            results=c("~/BTSync/FetalRNAseq/Counts/MvsF_all2",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_old",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_14_20A",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_12_14_noA",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_LRT_12_20_noA",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_LRT_14_20_noA",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks2",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks1",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks.75",
                                      "~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_16428"
                                      )
                            )
kable(select(DESeq2results, -results) %>% filter(Excluded == "None"))
```

# Dealing with outliers
`r margin_note("- Cooks Distance is a measure of the influence of each datapoints on the model fit")`
`r margin_note("- Cooks = 100 should read 100+ and Count = 10.0 should read 10+")`
`r margin_note("- Cooks Distance is << 1 for the vast majority of datapoints, but 15641, 18432 and possibly 16428 stand out as having many high values")`

![Distribution of Cooks Distances by Sample](/Users/heo3/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA/figures/CooksHist.png)

`r margin_note("- removing 15641 reduces the number of DE genes by 14% while removing 15641 and 18432 actually increases the number of DE genes by 11%")`
`r margin_note("- also removing 16428 reduces the number of DE 60%")`
`r margin_note("- This is presumably due to a reduction in variance when the outliers are removed")`

```{r}
kable(select(DESeq2results, -results) %>% 
        filter(BrainBank ==  "HDBR only" & AgeRange == "14-20" & test == 'Wald' & CooksCutoff == "None")
      )
```

# Filtering out genes that include samples with high Cooks Distance
`r margin_note("- removing samples with Cooks values > 2 reduces the number of DE genes by 40%")`
`r margin_note("- Most of these are likely to be spurious results")`
`r margin_note("- More stringent filtering actually increases the number of DE genes slightly, presumably because it is applied before FD correction")`

```{r}
kable(select(DESeq2results, -results) %>% 
        filter(n == 51)
      )
```

```{r}
PlotExpression<-function(geneID, file=All_Samples, geneName='', id="A value that hopefully isn't in the dataset", maxCooks=NULL) {
  if ( is.null(maxCooks) ) {
    maxCooks <-filter(file, Id==geneID)$maxCooks
  }    
  data <- filter(file, Id == geneID) %>%  
    select(starts_with('norm')) %>%
    gather() %>%
    separate(key, into=c('norm', 'label'), sep='[.]') %>%
    select(label, value) %>%
    left_join(target, by='label')
  plot<-  ggplot(data, aes(x=PCW, y=value, colour=Sex)) + 
      geom_point() + 
      geom_point(data=subset(data, label==id), colour='orange') +
      geom_smooth() +
      ylab("normalised counts") +
      tufte_theme() +
      scale_colour_brewer(type = "qual", palette = 6) +
      ggtitle(paste(geneName, "maxCooks = ", maxCooks)) +
      theme(legend.position=c(0.1,.9)) +
      theme(plot.background=element_blank())
    plot
}
GetID <- function(file, row_num) {
  read.delim(file, stringsAsFactors = FALSE) %>% 
    arrange(desc(maxCooks)) %>%
    select(Id, maxCooks) %>% 
    .[row_num,]
}
```

```{r }
targetFile <- "~/BTSync/FetalRNAseq/LabNotes/MvsFmac.txt"
target <- read.delim(targetFile, stringsAsFactors = FALSE)     # path to the design/target file

sample_info <- read.delim("~/BTSync/FetalRNAseq/LabNotes/sample_info.txt", stringsAsFactors = FALSE)
target <- left_join(target, select(sample_info, BrainBankID, Sex, PCW, RIN), by = c("label" = "BrainBankID"))

sample_progress <- read.delim("~/BTSync/FetalRNAseq/LabNotes/SampleProgress.txt", stringsAsFactors = FALSE)
target <- left_join(target, select(sample_progress, sample, Centre), by = c("label" = "sample"))
target <- arrange(target, Sex)

All_Samples <- read.delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA/tables/MalevsFemale.complete.txt", stringsAsFactors = FALSE)

FinalSamples <- read.delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks2/tables/MalevsFemale.complete.txt", stringsAsFactors = FALSE)
```

#Plot counts for gene with the highest Cooks Distances when cutoff is 2
`r margin_note("- Samples with Cooks Distance between 1 and 2 have an obvious impact on the model fit")`

```{r}
topCooksMale <- GetID("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks2/tables/MalevsFemale.up.txt", 1)
PlotExpression(topCooksMale[,1], All_Samples, '', maxCooks=topCooksMale[,2])

topCooksFemale <- GetID("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks2/tables/MalevsFemale.down.txt", 1)
PlotExpression(topCooksFemale[,1], All_Samples, '', maxCooks=topCooksFemale[,2])
```

#Plot counts for gene with the highest Cooks Distances when cutoff is 1
`r margin_note("- Samples with Cooks Distance under 1 still have noticible impact on the model fit")`

```{r}
topCooksMale <- GetID("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks1/tables/MalevsFemale.up.txt", 1)
PlotExpression(topCooksMale[,1], All_Samples, '', maxCooks=topCooksMale[,2])

topCooksFemale <- GetID("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks1/tables/MalevsFemale.down.txt", 1)
PlotExpression(topCooksFemale[,1], All_Samples, '', maxCooks=topCooksFemale[,2])
```
#Plot counts for gene with the highest Cooks Distances when cutoff is 0.75
`r margin_note("- Samples with Cooks Distance under 0.75 still stand out as outliers, but no longer appear to be driving the DE")`

```{r}
topCooksMale <- GetID("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks.75/tables/MalevsFemale.up.txt", 1)
PlotExpression(topCooksMale[,1], All_Samples, '', maxCooks=topCooksMale[,2])

topCooksFemale <- GetID("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks.75/tables/MalevsFemale.down.txt", 1)
PlotExpression(topCooksFemale[,1], All_Samples, '', maxCooks=topCooksFemale[,2])
```

