---
title: "DESeq2 Results"
author: "Heath O'Brien"
date: "`r Sys.Date()`"
output:
  tufte::tufte_html: default
---
<!-- see http://rstudio.github.io/tufte/ for info about tufte -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warnings=FALSE, message = FALSE, dev.args = list(bg = 'transparent'))

library(readr)
library(dplyr)
library(tidyr)
library(knitr)
library(ggplot2)
library(tufte)
library(gplots)
source("~/BTSync/FetalRNAseq/LabNotes/R/FormatGGplot.R")
#rm(list=ls()) 
```

# Running DEseq2 via SARTools (at FDR < 0.05)

`r margin_note("- All samples <14 weeks are from HDBR and all samples >20 weeks are not")`
`r margin_note("- Model for Wald tests is ~ Sex + PCW + RIN + Centre")`
`r margin_note("- LRT is ~ Sex * PCW + RIN + Centre vs. ~ PCW + RIN + Centre")`
`r margin_note("- Very few DE genes when samples > 14 weeks included, and most are on sex chromosomes")`
`r margin_note("- Excluding non HDBR samples increases the number of DE genes")`
`r margin_note("- LRT results in slightly more DE genes when analysing all HDBR samples, but far less when analysing samples > 14 PCW")`
`r margin_note("- For some reason, when I reran the 12-20 week anaysis, the number of DE genes dropped from 105 to 88 (old results  are in MvsF_12_20_noA_bk)")`
`r margin_note("- Excluding 3 female samples >= 18 weeks reduces the number of DE genes dramatically")`

```{r}
DESeq2results <- read_delim("~/BTSync/FetalRNAseq/Counts/Summary.txt", 
                            "\t", 
                            escape_double = FALSE, 
                            trim_ws = TRUE
                            )
filter(DESeq2results, grepl('new', results)) %>% select( -results, -test, -RIN, -model) %>% arrange(BrainBank, AgeRange, desc(Excluded)) %>% kable()
```

# Dealing with outliers
`r margin_note("- Cooks Distance is a measure of the influence of each datapoints on the model fit")`
`r margin_note("- Cooks = 100 should read 100+ and Count = 10.0 should read 10+")`
`r margin_note("- Cooks Distance is << 1 for the vast majority of datapoints, but 15641, 18432 and possibly 16428 stand out as having many high values")`

![Distribution of Cooks Distances by Sample](/Users/heo3/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_new/figures/CooksHist.png)
 
 
# Compare DESeq with EdgeR
`r margin_note("- Differences between 12-14/12-20 and 14-20 are more dramatic with edgeR")`
## 12-20
```{r}
MalevsFemale_up_12_20_EdgeR <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA_excl_15641_18432_16491_FDR.1_edgeR/tables/MalevsFemale.up.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_up_12_20_Deseq <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA_Cooks.75_excl_15641_18432_16491_FDR.1_new/tables/MalevsFemale.up.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_up_12_20_Voom <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA_excl_15641_18432_16491_FDR.1_voom/tables/MaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)

MalevsFemale_down_12_20_EdgeR <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA_excl_15641_18432_16491_FDR.1_edgeR/tables/MalevsFemale.down.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_down_12_20_Deseq <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA_Cooks.75_excl_15641_18432_16491_FDR.1_new/tables/MalevsFemale.down.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_down_12_20_Voom <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA_excl_15641_18432_16491_FDR.1_voom/tables/FemaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)

par(mfrow=c(1,2))
venn(list(DESeq=MalevsFemale_down_12_20_Deseq$Id, edgeR=MalevsFemale_down_12_20_EdgeR$Id, Voom=MalevsFemale_down_12_20_Voom$Id))
title(main="MalevsFemale_down FDR<0.1")
venn(list(DESeq=MalevsFemale_up_12_20_Deseq$Id, edgeR=MalevsFemale_up_12_20_EdgeR$Id,  Voom=MalevsFemale_up_12_20_Voom$Id))
title(main="MalevsFemale_up FDR<0.1")

MalevsFemale_up_12_20_EdgeR <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA_excl_15641_18432_16491_edgeR/tables/MalevsFemale.up.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_up_12_20_Deseq <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA_Cooks.75_excl_15641_18432_16491_new/tables/MalevsFemale.up.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_down_12_20_EdgeR <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA_excl_15641_18432_16491_edgeR/tables/MalevsFemale.down.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_down_12_20_Deseq <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA_Cooks.75_excl_15641_18432_16491_new/tables/MalevsFemale.down.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
par(mfrow=c(1,2))
venn(list(DESeq=MalevsFemale_down_12_20_Deseq$Id, edgeR=MalevsFemale_down_12_20_EdgeR$Id))
title(main="MalevsFemale_down FDR<0.05")
venn(list(DESeq=MalevsFemale_up_12_20_Deseq$Id, edgeR=MalevsFemale_up_12_20_EdgeR$Id))
title(main="MalevsFemale_up FDR<0.05")

```

## 12-14
```{r}
MalevsFemale_up_12_14_EdgeR <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_14_noA_excl_16491_FDR.1_edgeR/tables/MalevsFemale.up.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_up_12_14_Deseq <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_14_noA_Cooks.75_excl_16491_FDR.1_new/tables/MalevsFemale.up.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_down_12_14_EdgeR <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_14_noA_excl_16491_FDR.1_edgeR/tables/MalevsFemale.down.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_down_12_14_Deseq <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_14_noA_Cooks.75_excl_16491_FDR.1_new/tables/MalevsFemale.down.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
par(mfrow=c(1,2))
venn(list(DESeq=MalevsFemale_down_12_14_Deseq$Id, edgeR=MalevsFemale_down_12_14_EdgeR$Id))
title(main="MalevsFemale_down FDR<0.1")
venn(list(DESeq=MalevsFemale_up_12_14_Deseq$Id, edgeR=MalevsFemale_up_12_14_EdgeR$Id))
title(main="MalevsFemale_up FDR<0.1")

MalevsFemale_up_12_14_EdgeR <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_14_noA_excl_16491_edgeR/tables/MalevsFemale.up.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_up_12_14_Deseq <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_14_noA_Cooks.75_excl_16491_new/tables/MalevsFemale.up.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_down_12_14_EdgeR <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_14_noA_excl_16491_edgeR/tables/MalevsFemale.down.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
MalevsFemale_down_12_14_Deseq <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_14_noA_Cooks.75_excl_16491_new/tables/MalevsFemale.down.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
par(mfrow=c(1,2))
venn(list(DESeq=MalevsFemale_down_12_14_Deseq$Id, edgeR=MalevsFemale_down_12_14_EdgeR$Id))
title(main="MalevsFemale_down FDR<0.05")
venn(list(DESeq=MalevsFemale_up_12_14_Deseq$Id, edgeR=MalevsFemale_up_12_14_EdgeR$Id))
title(main="MalevsFemale_up FDR<0.05")
```

## 14-20
```{r}
Male_up_14_20_EdgeR <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_FDR.1_edgeR/tables/MaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
Male_up_14_20_Deseq <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_Cooks.75_excl_15641_18432_FDR.1_new/tables/MaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
Male_up_14_20_Voom <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_voom/tables/MaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)

Female_up_14_20_EdgeR <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_FDR.1_edgeR/tables/FemaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
Female_up_14_20_Deseq <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_Cooks.75_excl_15641_18432_FDR.1_new/tables/FemaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
Female_up_14_20_Voom <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_voom/tables/FemaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
par(mfrow=c(1,2))
venn(list(DESeq=Male_up_14_20_Deseq$Id, edgeR=Male_up_14_20_EdgeR$Id, Voom=Male_up_14_20_Voom$Id))
title(main="Male Up FDR<0.1")
venn(list(DESeq=Female_up_14_20_Deseq$Id, edgeR=Female_up_14_20_EdgeR$Id, Voom=Female_up_14_20_Voom$Id))
title(main="Female Up FDR<0.1")

Female_up_intersection <- inner_join(select(Female_up_14_20_EdgeR, Chr, GeneID, Id, padj), select(Female_up_14_20_Deseq, Id, padj), by='Id')
colnames(Female_up_intersection) <- c("Chr", "GeneID", "Id", "padj_EdgeR", "padj_DEseq")
write_delim(Female_up_intersection, "~/BTSync/FetalRNAseq/Counts/MvsF_14_20_female_up_intersection.txt", delim='\t')

Male_up_intersection <- inner_join(select(Male_up_14_20_EdgeR, Chr, GeneID, Id, padj), select(Male_up_14_20_Deseq, Id, padj), by='Id')
colnames(Male_up_intersection) <- c("Chr", "GeneID", "Id", "padj_EdgeR", "padj_DEseq")
write_delim(Male_up_intersection, "~/BTSync/FetalRNAseq/Counts/MvsF_14_20_male_up_intersection.txt", delim='\t')

Male_up_14_20_FDR.05_EdgeR <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_edgeR/tables/MaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
Male_up_14_20_FDR.05_Deseq <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_Cooks.75_excl_15641_18432_new/tables/MaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
Male_up_14_20_FDR.05_Voom <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_FDR.05_voom/tables/MaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)

Female_up_14_20_FDR.05_EdgeR <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_edgeR/tables/FemaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
Female_up_14_20_FDR.05_Deseq <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_Cooks.75_excl_15641_18432_new/tables/FemaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
Female_up_14_20_FDR.05_Voom <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_FDR.05_voom/tables/FemaleUp.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)

par(mfrow=c(1,2))
venn(list(DESeq=Male_up_14_20_FDR.05_Deseq$Id, edgeR=Male_up_14_20_FDR.05_EdgeR$Id, Voom=Male_up_14_20_FDR.05_Voom$Id))
title(main="Male Up FDR<0.05")
venn(list(DESeq=Female_up_14_20_FDR.05_Deseq$Id, edgeR=Female_up_14_20_FDR.05_EdgeR$Id, Voom=Female_up_14_20_FDR.05_Voom$Id))
title(main="Female Up FDR<0.05")
```



```{r }
targetFile <- "~/BTSync/FetalRNAseq/LabNotes/MvsFmac.txt"
target <- read.delim(targetFile, stringsAsFactors = FALSE)     # path to the design/target file

sample_info <- read.delim("~/BTSync/FetalRNAseq/LabNotes/sample_info.txt", stringsAsFactors = FALSE)
target <- left_join(target, select(sample_info, BrainBankID, Sex, PCW, RIN), by = c("label" = "BrainBankID"))

sample_progress <- read.delim("~/BTSync/FetalRNAseq/LabNotes/SampleProgress.txt", stringsAsFactors = FALSE)
target <- left_join(target, select(sample_progress, sample, Centre), by = c("label" = "sample"))
target <- arrange(target, Sex)
target <- mutate(target, PCW = factor(floor(PCW)))
#All_Samples <- read.delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA/tables/MalevsFemale.complete.txt", stringsAsFactors = FALSE)
```

# Genes of interest

```{r}
my_db <- src_mysql("FetalRNAseq", host="localhost", user="root")
GetGeneID <- function(gene_name) {
  geneID <- collect(tbl(my_db, 
                        sql(paste0("SELECT gene_id FROM GencodeGenes WHERE gene_name = '",
                                   gene_name, 
                                   "'"
                                   )
                            )
                        )) 
  as.character(geneID[1,1])
}

NoteSig <- function(gene_id, gene_name, EarlySamples, LateSamples, All_Samples) {
  pvalues <-filter(EarlySamples, Id==gene_id) %>% 
    select(FoldChange, pvalue, padj) %>% 
    mutate(Gene=gene_name, 
                       Age='12-14 weeks', 
                       FoldChange=prettyNum(FoldChange, digits=3),
                       pvalue=prettyNum(pvalue, digits=2),
                       padj=prettyNum(padj, digits=2)
           ) %>%
    bind_rows(filter(LateSamples, Id==gene_id) %>% 
                select(FoldChange, pvalue, padj) %>% 
                mutate(Gene=gene_name, 
                       Age='14-20 weeks', 
                       FoldChange=prettyNum(FoldChange, digits=3),
                       pvalue=prettyNum(pvalue, digits=2),
                       padj=prettyNum(padj, digits=2)
                       )
              ) %>%
    bind_rows(filter(All_Samples, Id==gene_id) %>% 
                select(FoldChange, pvalue, padj) %>% 
                mutate(Gene=gene_name, 
                       Age='dev', 
                       FoldChange=prettyNum(FoldChange, digits=3),
                       pvalue=prettyNum(pvalue, digits=2),
                       padj=prettyNum(padj, digits=2)
                       )
              )
  pvalues[,c(3,4,5,1,2)]
}
InitializePvalues <- function() {
  data.frame(Gene=as.character(), 
                      Age=as.character(), 
                      pvalue=as.character(), 
                      padj=as.character(),
                      FoldChange=as.character(),
                      stringsAsFactors = FALSE)
}
```

In males, there is significantly (P < 0.05, uncorrected) higher expression (relative to females) of key markers of basal neural progenitor cells  (PAX6, SOX2 and EOMES, with PAX6 FDR< 0.05) after 14 weeks. (None of these differ significantly from females < 14 weeks.) 

```{r}
pvalues <- InitializePvalues()
for (gene_name in c('PAX6', 'SOX2', 'EOMES', 'CHRM2')) {
  gene_id <- GetGeneID(gene_name)

  print(PlotExpression(gene_id, All_Samples, gene_name))
  pvalues <- bind_rows(pvalues, NoteSig(gene_id, gene_name, EarlySamples, LateSamples, All_Samples))
}
kable(pvalues)
```

In females, there is significantly higher relative expression of a key marker of differentiated (post-mitotic) GABAergic neurons (GAD1, FDR < 0.01).  There is also significantly greater (although not FDR < 0.05) synaptophysin expression (a classic synaptic marker).

```{r}
pvalues <- InitializePvalues()
for (gene_name in c('GAD1', 'SYP')) {
  gene_id <- GetGeneID(gene_name)
  print(PlotExpression(gene_id, All_Samples, gene_name))
  pvalues <- bind_rows(pvalues, NoteSig(gene_id, gene_name, EarlySamples, LateSamples, All_Samples))
}
kable(pvalues)

```

After 14 weeks, there is also higher expression of a Type 2 astrocyte marker in males (MSI1, FDR < 0.05), whereas there is increased expression of markers of oligodendrocytes progenitors (OLIG1 FDR < 0.05, OLIG2 FDR = 0.06) and mature oligodendrocytes (GALC, FDR < 0.01) in females. My knowledge of glial cell development isn't so good, but I think oligodendrocytes again develop later (their main function is to myelinate neurites).

```{r}
pvalues <- InitializePvalues()
for (gene_name in c('MSI1', 'OLIG1', 'OLIG2', 'GALC')) {
  gene_id <- GetGeneID(gene_name)
  print(PlotExpression(gene_id, All_Samples, gene_name))
  pvalues <- bind_rows(pvalues, NoteSig(gene_id, gene_name, EarlySamples, LateSamples, All_Samples))
}
kable(pvalues)

```

... and there is also increased expression of the glutamatergic neuron marker vGluT2 (SLC17A6) in females after 4 weeks (FDR = ~0.01). So that's both neuronal subtypes of the cerebral cortex (Glu and GABA)!

```{r}
pvalues <- InitializePvalues()
for (gene_name in c('SLC17A6')) {
  gene_id <- GetGeneID(gene_name)
  print(PlotExpression(gene_id, All_Samples, gene_name))
  pvalues <- bind_rows(pvalues, NoteSig(gene_id, gene_name, EarlySamples, LateSamples, All_Samples))
}
kable(pvalues)
```

```{r}
pvalues <- InitializePvalues()
for (gene_name in c('NES', 'DCX', 'GFAP', 'MAP2', 'TUBB3', 'TH', 'FOXG1', 'TBR1', 'EMX1', 'ACTB')) {
  gene_id <- GetGeneID(gene_name)
  print(PlotExpression(gene_id, All_Samples, gene_name))
  pvalues <- bind_rows(pvalues, NoteSig(gene_id, gene_name, EarlySamples, LateSamples, All_Samples))
}
kable(pvalues)

```
```{r}
pvalues <- InitializePvalues()
for (gene_name in c('MYOT', 'IGLV4-69', 'NR2F6', 'TOPORS')) {
  gene_id <- GetGeneID(gene_name)
  print(PlotExpression(gene_id, All_Samples, gene_name))
  pvalues <- bind_rows(pvalues, NoteSig(gene_id, gene_name, EarlySamples, LateSamples, All_Samples))
}
kable(pvalues)
```

```{r}




```
```{r}
Week12 <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_13_noA_FDR.1_edgeR/tables/MalevsFemale.complete.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
Week13 <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_13_14_noA_FDR.1_edgeR/tables/MalevsFemale.complete.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
Week14 <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_15_noA_FDR.1_edgeR/tables/MalevsFemale.complete.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
Week15_16 <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_15_17_noA__excl_15641_Cooks.75_FDR.1_edgeR/tables/MalevsFemale.complete.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)
Week17_19 <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_17_20_noA__excl_18432_Cooks.75_FDR.1_edgeR/tables/MalevsFemale.complete.txt", 
     "\t", escape_double = FALSE, trim_ws = TRUE)

data <- full_join(mutate(Week12, Male_12=Male, Female_12=Female) %>%
                    select(starts_with('norm'), Id, Male_12, Female_12),
                   mutate(Week13, Male_13=Male, Female_13=Female) %>%
                    select(starts_with('norm'), Id, Male_13, Female_13)
                  ) %>%
        full_join(mutate(Week14, Male_14=Male, Female_14=Female) %>%
                    select(starts_with('norm'), Id, Male_14, Female_14) 
                  ) %>%
        full_join(mutate(Week15_16, Male_15.5=Male, Female_15.5=Female) %>%
                    select(starts_with('norm'), Id, Male_15.5, Female_15.5)
                  ) %>%
        full_join(mutate(Week17_19, Male_18=Male, Female_18=Female) %>%
                    select(starts_with('norm'), Id, Male_18, Female_18)
                 )

PlotCounts <- function(data, gene_name) {
  gene_id <- GetGeneID(gene_name)
  counts <- filter(data, Id == gene_id) %>% 
    select(starts_with('norm')) %>%
    gather() %>%
    separate(key, into=c('norm', 'label'), sep='[.]') %>%
    select(label, value) %>%
    left_join(target, by='label') %>%
    mutate(PCW=as.numeric(as.character(PCW)))
  counts$PCWpool <- ifelse(counts$PCW >= 15 & counts$PCW < 17, 15.5,   ifelse(counts$PCW >= 17 & counts$PCW < 20, 18, counts$PCW))

  fit <- filter(data, Id ==gene_id) %>%
         select(-starts_with('norm')) %>%
         gather(SexPCW, Fit, -Id) %>%
         separate(SexPCW, c('Sex', 'PCW'), sep='_') %>%
         mutate(PCW = as.numeric(as.character(PCW)))


  plot <- ggplot(counts, aes(x=PCWpool, y=log2(value), colour=Sex)) + 
  #plot <- ggplot(counts, aes(x=PCWpool, y=value, colour=Sex)) + 
            geom_jitter(height=0, width=.1) +
            geom_line(aes(x=PCW, y=log2(Fit), colour=Sex), data=fit) +
            #geom_line(aes(x=PCW, y=Fit, colour=Sex), data=fit) +
            scale_x_continuous(breaks=c(12,13,14,15.5,18), 
                               labels=c('12', '13', '14', '15-16', '17-19')) +
            scale_y_continuous(breaks=c(2,4,6,8,10,12,14,16,18,20), 
                               labels=c(2^2, 2^4,2^6,2^8,2^10,2^12,2^14,2^16,2^18,2^20)) +
            ylab("normalised counts") +
            xlab("Weeks Post Conception") +
            tufte_theme() +
            scale_colour_brewer(type = "qual", palette = 6) +
            theme(legend.position=c(0.75,.9)) +
            theme(plot.background=element_blank()) +
            ggtitle(gene_name)
  plot
}
```

```{r}
PlotCounts(data, 'XIST') #X
PlotCounts(data, 'BNC2')
PlotCounts(data, 'SHOX') #X
PlotCounts(data, 'NLGN4X') #X
PlotCounts(data, 'DTHD1')
PlotCounts(data, 'RP11-707M1.1')
PlotCounts(data, 'WDR63')
PlotCounts(data, 'ONECUT3')
```

```{r}
PlotCounts(data, 'USP9Y') #Y
PlotCounts(data, 'NFATC4')
PlotCounts(data, 'CHRDL1') #X
PlotCounts(data, 'NEUROD4')
PlotCounts(data, 'LRP4')
PlotCounts(data, 'NPR3')
PlotCounts(data, 'ESPL1')
PlotCounts(data, 'MCM2')
```
