---
title: "DESeq2 Results"
author: "Heath O'Brien"
subtitle: "An implementation in R Markdown"
date: "`r Sys.Date()`"
output:
  tufte::tufte_html: default
---
<!-- see http://rstudio.github.io/tufte/ for info about tufte -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warnings=FALSE, message = FALSE, dev.args = list(bg = 'transparent'))
rm(list=ls())

library(readr)
library(dplyr)
library(tidyr)
library(ggplot2)
source("~/BTSync/FetalRNAseq/LabNotes/R/FormatGGplot.R")
```

```{r}
PlotExpression<-function(geneID, file=All_Samples, geneName='', id="A value that hopefully isn't in the dataset") {
  maxCooks <-filter(file, Id==geneID)$maxCooks
  data <- filter(file, Id == geneID) %>%  
    select(starts_with('norm')) %>%
    gather() %>%
    separate(key, into=c('norm', 'label'), sep='[.]') %>%
    select(label, value) %>%
    left_join(target, by='label')
  plot<-  ggplot(data, aes(x=PCW, y=value, colour=Sex)) + 
      geom_point() + 
      geom_point(data=subset(data, label==id), colour='orange') +
      geom_smooth() +
      ylab("normalised counts") +
      tufte_theme() +
      scale_colour_brewer(type = "qual", palette = 6) +
      ggtitle(paste(geneName, "maxCooks = ", maxCooks)) +
      theme(legend.position=c(0.1,.9)) +
      theme(plot.background=element_blank())
    plot
}
```

```{r }
targetFile <- "~/BTSync/FetalRNAseq/LabNotes/MvsFmac.txt"
target <- read.delim(targetFile)                        # path to the design/target file

sample_info <- read.delim("~/BTSync/FetalRNAseq/LabNotes/sample_info.txt")
target <- left_join(target, select(sample_info, BrainBankID, Sex, PCW, RIN), by = c("label" = "BrainBankID"))

sample_progress <- read.delim("~/BTSync/FetalRNAseq/LabNotes/SampleProgress.txt")
target <- left_join(target, select(sample_progress, sample, Centre), by = c("label" = "sample"))
target <- arrange(target, Sex)

All_Samples <- read.delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA/tables/MalevsFemale.complete.txt")

FinalSamples <- read.delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks2/tables/MalevsFemale.complete.txt")
```

# Running DEseq2 via SARTools (at FDR < 0.05)
- Note that all 'A' samples are >14 weeks and all samples >20 weeks are labelled A
- When M vs. F Wald test run on all samples (N=108), there are 68 DE genes (mostly on sex chromosomes)
    - When samples labeled A are excluded (N=100), there are 105 DE genes
    - When samples < 14 weeks are excluded (N=61), there are 633 DE genes
    - When samples < 14 and >= 20 weeks are excluded (N=58), there are 1747 DE genes
        - When samples labeled A are excluded (N=53), there are 2665 DE genes

        
    - When samples >= 14 weeks are excluded (N=47), there are 47 DE genes
- When LRT is run on all samples not labeled A (N=100), using Sex * Age in the full model and dropping Sex in the reduced model, there are 154 DE genes
    - When samples < 14 and >= 20 weeks and labeled A are excluded (N=53), there are 868 DE genes

When the two samples with the most high Cooks distances (15641 and 18432), the number of DE genes actually increases to 2961

When genes with a high maximum Cooks distance (>2) are filtered out, the number of DE genes drops to 1738

This still includes genes where a single sample has a large impact on the results
```{r}
PlotExpression('ENSG00000163519.13', FinalSamples, '')
PlotExpression('ENSG00000244056.3', FinalSamples, '')
```

Dropping the cutoff to 1 actually increases the number of DE genes slightly to 1763 (presumably because the number of comparasons used in the FDR correction is reduced )

The outliers are still very obvious, but it no longer looks like they are driving all of the differences
```{r}
PlotExpression('ENSG00000257531.2', FinalSamples, '')
PlotExpression('ENSG00000139865.16', FinalSamples, '')
```

Dropping the cutoff to .75 decreases the number of DE genes very slightly to 1756 

At this cutoff, there is still a lot of spread, but nothing identifiable as an outlier
```{r}
PlotExpression('ENSG00000170017.12', FinalSamples, '')
PlotExpression('ENSG00000231535.5', FinalSamples, '')
```


```{r}
PlotExpression('ENSG00000049323.15', All_Samples, 'LTPB1')
```

```{r}
PlotExpression('ENSG00000254338.1', All_Samples, 'MAFA-AS1') # not so clear
```
```{r}
PlotExpression('ENSG00000204054.11', All_Samples, 'LINC00963') # not so clear
```
```{r}
PlotExpression('ENSG00000102967.11', All_Samples, 'DHODH')
```
```{r}
PlotExpression('ENSG00000105137.12', All_Samples, 'SYDE1')
```
```{r}
PlotExpression('ENSG00000077935.16', All_Samples, 'SMC1B', '15641')
```
```{r}
PlotExpression('ENSG00000218536.1', All_Samples, 'AP002530.2', '19052')
```
```{r}
PlotExpression('ENSG00000187416.11', All_Samples, 'LHFPL3')
```
```{r}
PlotExpression('ENSG00000233593.6', All_Samples, 'RP4-665J23.1', '19052')
```

```{r}
PlotExpression('ENSG00000170537.12', All_Samples, 'TMC7')
```

Genes that have a significant sex-by-age interaction are due to outliers at age extremes

nbinomLRT(dds, 
          full=formula('~ Sex * PCW + Centre + RIN'), 
          reduced=formula('~ Sex + PCW + Centre + RIN')
          )
          
```{r dev.args = list(bg = 'transparent')}
PlotExpression('ENSG00000173714.7', All_Samples)
```
```{r}
PlotExpression('ENSG00000187783.11', All_Samples)
```

```{r}
PlotExpression('ENSG00000165125.17', All_Samples)
```

```{r}
PlotExpression('ENSG00000111405.8', All_Samples)
```

```{r}
PlotExpression('ENSG00000118271.9', All_Samples)
```

```{r}
#PlotExpression('ENSG00000151962.7', 'RBM46', MalevsFemale.down, '15641')
#PlotExpression('ENSG00000131914.10', 'LIN28A', MalevsFemale.down, '15641')
#PlotExpression('ENSG00000205089.7', 'CCNI2', MalevsFemale.down, '15641')
#PlotExpression('ENSG00000086548.8', 'CEACAM6', MalevsFemale.down, '17812')
#PlotExpression('ENSG00000238269.8', 'PAGE2B', MalevsFemale.down, '15641')
#PlotExpression('ENSG00000260822.1', 'GS1-358P8.4', MalevsFemale.down)
#PlotExpression('ENSG00000167077.12', 'MEI1', MalevsFemale.down, '15641')
#PlotExpression('ENSG00000102837.6', 'OLMF4', MalevsFemale.down, '17812')
#PlotExpression('ENSG00000001461.16', 'NIPAL3', MalevsFemale.down)
#PlotExpression('ENSG00000185053.12', 'SGCZ', MalevsFemale.down)
#PlotExpression('ENSG00000249937.5', 'RP11-454P21.1', MalevsFemale.down)
#PlotExpression('ENSG00000240032.1', 'RP11-274H2.3', MalevsFemale.down)
#PlotExpression('ENSG00000159374.17', 'M1AP', MalevsFemale.down, '15641')
#PlotExpression('ENSG00000159110.19', 'IFNAR2', MalevsFemale.down)


MalevsFemale.down <- read_tsv("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA/tables/MalevsFemale.down.txt",
                             col_types =cols_only(maxCooks='n')
                             )
MalevsFemale.up <- read_tsv("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA/tables/MalevsFemale.up.txt",
                             col_types =cols_only(maxCooks='n')
                             )
maxCooksDF <- bind_rows(data.frame(maxCooks=MalevsFemale.down$maxCooks, Direction="FemaleUp"), data.frame(maxCooks=MalevsFemale.up$maxCooks, Direction="MaleUp"))
```

```{r}
ggplot(maxCooksDF, aes(x=maxCooks)) + 
  geom_histogram() +
  scale_x_log10() +
  tufte_theme() +
  facet_grid(Direction ~ .)

```
