---
title: "DESeq2 Results"
author: "Heath O'Brien"
date: "`r Sys.Date()`"
output:
  tufte::tufte_html: default
---
<!-- see http://rstudio.github.io/tufte/ for info about tufte -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warnings=FALSE, message = FALSE, dev.args = list(bg = 'transparent'))

library(readr)
library(dplyr)
library(tidyr)
library(knitr)
library(ggplot2)
library(tufte)
source("~/BTSync/FetalRNAseq/LabNotes/R/FormatGGplot.R")
```

# Running DEseq2 via SARTools (at FDR < 0.05)

`r margin_note("- All samples <14 weeks are from HDBR and all samples >20 weeks are not")`
`r margin_note("- Model for Wald tests is ~ Sex + PCW + RIN + Centre")`
`r margin_note("- LRT is ~ Sex * PCW + RIN + Centre vs. ~ PCW + RIN + Centre")`
`r margin_note("- Very few DE genes when samples > 14 weeks included, and most are on sex chromosomes")`
`r margin_note("- Excluding non HDBR samples increases the number of DE genes")`
`r margin_note("- LRT results in slightly more DE genes when analysing all HDBR samples, but far less when analysing samples > 14 PCW")`
`r margin_note("- For some reason, when I reran the 12-20 week anaysis, the number of DE genes dropped from 105 to 88 (old results  are in MvsF_12_20_noA_bk)")`
`r margin_note("- Excluding 3 female samples >= 18 weeks reduces the number of DE genes dramatically")`

```{r}
DESeq2results <- read_delim("~/BTSync/FetalRNAseq/Counts/Summary.txt", 
                            "\t", 
                            escape_double = FALSE, 
                            trim_ws = TRUE
                            )
filter(DESeq2results, grepl('new', results)) %>% select( -results, -test, -RIN, -model) %>% arrange(BrainBank, AgeRange, desc(Excluded)) %>% kable()
```

# Dealing with outliers
`r margin_note("- Cooks Distance is a measure of the influence of each datapoints on the model fit")`
`r margin_note("- Cooks = 100 should read 100+ and Count = 10.0 should read 10+")`
`r margin_note("- Cooks Distance is << 1 for the vast majority of datapoints, but 15641, 18432 and possibly 16428 stand out as having many high values")`

![Distribution of Cooks Distances by Sample](/Users/heo3/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_new/figures/CooksHist.png)
 
 
`r margin_note("- removing 15641 reduces the number of DE genes by 14% while removing 15641 and 18432 actually increases the number of DE genes by 11%")`
`r margin_note("- also removing 16428 reduces the number of DE 60%")`
`r margin_note("- This is presumably due to a reduction in variance when the outliers are removed")`
 
 
```{r}
kable(select(DESeq2results, -results) %>% 
        filter(BrainBank ==  "HDBR" & AgeRange == "14-20" & test == 'Wald' & CooksCutoff == "FALSE")
      )
```


# Filtering out genes that include samples with high Cooks Distance
`r margin_note("- removing samples with Cooks values > 2 reduces the number of DE genes by 40%")`
`r margin_note("- Most of these are likely to be spurious results")`
`r margin_note("- More stringent filtering actually increases the number of DE genes slightly, presumably because it is applied before FD correction")`

```{r}
kable(select(DESeq2results, -results) %>% 
        filter(n == 51)
      )
```

```{r}
PlotExpression<-function(geneID, file=All_Samples, geneName='', id="A value that hopefully isn't in the dataset", maxCooks=NULL) {
  if ( is.null(maxCooks) ) {
    maxCooks <-filter(file, Id==geneID)$maxCooks
  }    
  data <- filter(file, Id == geneID) %>%  
    select(starts_with('norm')) %>%
    gather() %>%
    separate(key, into=c('norm', 'label'), sep='[.]') %>%
    select(label, value) %>%
    left_join(target, by='label')
  plot<-  ggplot(data, aes(x=as.numeric(as.character(PCW)), y=value, colour=Sex)) + 
      geom_point() + 
      geom_point(data=subset(data, label==id), colour='orange') +
      geom_smooth() +
      ylab("normalised counts") +
      tufte_theme() +
      scale_colour_brewer(type = "qual", palette = 6) +
      ggtitle(paste(geneName, "maxCooks = ", maxCooks)) +
      theme(legend.position=c(0.1,.9)) +
      theme(plot.background=element_blank())
    plot
}

GetID <- function(file, row_num) {
  read.delim(file, stringsAsFactors = FALSE) %>% 
    arrange(desc(maxCooks)) %>%
    select(Id, maxCooks) %>% 
    .[row_num,]
}
```

```{r }
targetFile <- "~/BTSync/FetalRNAseq/LabNotes/MvsFmac.txt"
target <- read.delim(targetFile, stringsAsFactors = FALSE)     # path to the design/target file

sample_info <- read.delim("~/BTSync/FetalRNAseq/LabNotes/sample_info.txt", stringsAsFactors = FALSE)
target <- left_join(target, select(sample_info, BrainBankID, Sex, PCW, RIN), by = c("label" = "BrainBankID"))

sample_progress <- read.delim("~/BTSync/FetalRNAseq/LabNotes/SampleProgress.txt", stringsAsFactors = FALSE)
target <- left_join(target, select(sample_progress, sample, Centre), by = c("label" = "sample"))
target <- arrange(target, Sex)
target <- mutate(target, PCW = factor(floor(PCW)))
#All_Samples <- read.delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA/tables/MalevsFemale.complete.txt", stringsAsFactors = FALSE)

All_Samples <- read.delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_noA_new/tables/MalevsFemale.complete.txt", stringsAsFactors = FALSE)

LateSamples <- read.delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_Cooks.75_excl_15641_18432_new/tables/MalevsFemale.complete.txt", stringsAsFactors = FALSE)

EarlySamples <- read.delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_14_noA_Cooks.75_excl_16491_new/tables/MalevsFemale.complete.txt", stringsAsFactors = FALSE)
```

#Plot counts for gene with the highest Cooks Distances when cutoff is 2
`r margin_note("- Samples with Cooks Distance between 1 and 2 have an obvious impact on the model fit")`

```{r}
topCooksMale <- GetID("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks2/tables/MalevsFemale.up.txt", 1)
PlotExpression(topCooksMale[,1], All_Samples, '', maxCooks=topCooksMale[,2])

topCooksFemale <- GetID("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks2/tables/MalevsFemale.down.txt", 1)
PlotExpression(topCooksFemale[,1], All_Samples, '', maxCooks=topCooksFemale[,2])
```

#Plot counts for gene with the highest Cooks Distances when cutoff is 1
`r margin_note("- Samples with Cooks Distance under 1 still have noticible impact on the model fit")`

```{r}
topCooksMale <- GetID("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks1/tables/MalevsFemale.up.txt", 1)
PlotExpression(topCooksMale[,1], All_Samples, '', maxCooks=topCooksMale[,2])

topCooksFemale <- GetID("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks1/tables/MalevsFemale.complete.txt", 1)
PlotExpression(topCooksFemale[,1], All_Samples, '', maxCooks=topCooksFemale[,2])
```
#Plot counts for gene with the highest Cooks Distances when cutoff is 0.75
`r margin_note("- Samples with Cooks Distance under 0.75 still stand out as outliers, but no longer appear to be driving the DE")`

```{r}
topCooksMale <- GetID("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks.75/tables/MalevsFemale.up.txt", 1)
PlotExpression(topCooksMale[,1], All_Samples, '', maxCooks=topCooksMale[,2])

topCooksFemale <- GetID("~/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA_excl_15641_18432_Cooks.75/tables/MalevsFemale.down.txt", 1)
PlotExpression(topCooksFemale[,1], All_Samples, '', maxCooks=topCooksFemale[,2])
```

![Distribution of Cooks Distances by Sample (14-20 PCW)](/Users/heo3/BTSync/FetalRNAseq/Counts/MvsF_14_20_noA/figures/CooksHist.png)

`r margin_note("- removing 15641 reduces the number of DE genes by 14% while removing 15641 and 18432 actually increases the number of DE genes by 11%")`
`r margin_note("- also removing 16428 reduces the number of DE 60%")`
`r margin_note("- This is presumably due to a reduction in variance when the outliers are removed")`

# Outliers in 12-14 PCW samples

![Distribution of Cooks Distances by Sample (12-14 PCW)](/Users/heo3/BTSync/FetalRNAseq/Counts/MvsF_12_14_noA/figures/CooksHist.png)
`r margin_note("- DE analysis is robust to the removal of 16491 (46 DE genes vs 46 or 47 with, depending on run) and maxCooks filtering (48genes)")`

```{r}
kable(select(DESeq2results, -results) %>% 
        filter(BrainBank ==  "HDBR" & AgeRange == "12-14" & test == 'Wald')
      )
```

# Genes of interest

```{r}
my_db <- src_mysql("FetalRNAseq", host="localhost", user="root")
GetGeneID <- function(gene_name) {
  geneID <- collect(tbl(my_db, 
                        sql(paste0("SELECT gene_id FROM GencodeGenes WHERE gene_name = '",
                                   gene_name, 
                                   "'"
                                   )
                            )
                        )) 
  as.character(geneID[1,1])
}

NoteSig <- function(gene_id, gene_name, EarlySamples, LateSamples, All_Samples) {
  pvalues <-filter(EarlySamples, Id==gene_id) %>% 
    select(FoldChange, pvalue, padj) %>% 
    mutate(Gene=gene_name, 
                       Age='12-14 weeks', 
                       FoldChange=prettyNum(FoldChange, digits=3),
                       pvalue=prettyNum(pvalue, digits=2),
                       padj=prettyNum(padj, digits=2)
           ) %>%
    bind_rows(filter(LateSamples, Id==gene_id) %>% 
                select(FoldChange, pvalue, padj) %>% 
                mutate(Gene=gene_name, 
                       Age='14-20 weeks', 
                       FoldChange=prettyNum(FoldChange, digits=3),
                       pvalue=prettyNum(pvalue, digits=2),
                       padj=prettyNum(padj, digits=2)
                       )
              ) %>%
    bind_rows(filter(All_Samples, Id==gene_id) %>% 
                select(FoldChange, pvalue, padj) %>% 
                mutate(Gene=gene_name, 
                       Age='dev', 
                       FoldChange=prettyNum(FoldChange, digits=3),
                       pvalue=prettyNum(pvalue, digits=2),
                       padj=prettyNum(padj, digits=2)
                       )
              )
  pvalues[,c(3,4,5,1,2)]
}
InitializePvalues <- function() {
  data.frame(Gene=as.character(), 
                      Age=as.character(), 
                      pvalue=as.character(), 
                      padj=as.character(),
                      FoldChange=as.character(),
                      stringsAsFactors = FALSE)
}
```

In males, there is significantly (P < 0.05, uncorrected) higher expression (relative to females) of key markers of basal neural progenitor cells  (PAX6, SOX2 and EOMES, with PAX6 FDR< 0.05) after 14 weeks. (None of these differ significantly from females < 14 weeks.) 

```{r}
pvalues <- InitializePvalues()
for (gene_name in c('PAX6', 'SOX2', 'EOMES')) {
  gene_id <- GetGeneID(gene_name)

  print(PlotExpression(gene_id, All_Samples, gene_name))
  pvalues <- bind_rows(pvalues, NoteSig(gene_id, gene_name, EarlySamples, LateSamples, All_Samples))
}
kable(pvalues)
```

In females, there is significantly higher relative expression of a key marker of differentiated (post-mitotic) GABAergic neurons (GAD1, FDR < 0.01).  There is also significantly greater (although not FDR < 0.05) synaptophysin expression (a classic synaptic marker).

```{r}
pvalues <- InitializePvalues()
for (gene_name in c('GAD1', 'SYP')) {
  gene_id <- GetGeneID(gene_name)
  print(PlotExpression(gene_id, All_Samples, gene_name))
  pvalues <- bind_rows(pvalues, NoteSig(gene_id, gene_name, EarlySamples, LateSamples, All_Samples))
}
kable(pvalues)

```

After 14 weeks, there is also higher expression of a Type 2 astrocyte marker in males (MSI1, FDR < 0.05), whereas there is increased expression of markers of oligodendrocytes progenitors (OLIG1 FDR < 0.05, OLIG2 FDR = 0.06) and mature oligodendrocytes (GALC, FDR < 0.01) in females. My knowledge of glial cell development isn't so good, but I think oligodendrocytes again develop later (their main function is to myelinate neurites).

```{r}
pvalues <- InitializePvalues()
for (gene_name in c('MSI1', 'OLIG1', 'OLIG2', 'GALC')) {
  gene_id <- GetGeneID(gene_name)
  print(PlotExpression(gene_id, All_Samples, gene_name))
  pvalues <- bind_rows(pvalues, NoteSig(gene_id, gene_name, EarlySamples, LateSamples, All_Samples))
}
kable(pvalues)

```

... and there is also increased expression of the glutamatergic neuron marker vGluT2 (SLC17A6) in females after 4 weeks (FDR = ~0.01). So that's both neuronal subtypes of the cerebral cortex (Glu and GABA)!

```{r}
pvalues <- InitializePvalues()
for (gene_name in c('SLC17A6')) {
  gene_id <- GetGeneID(gene_name)
  print(PlotExpression(gene_id, All_Samples, gene_name))
  pvalues <- bind_rows(pvalues, NoteSig(gene_id, gene_name, EarlySamples, LateSamples, All_Samples))
}
kable(pvalues)
```

```{r}
pvalues <- InitializePvalues()
for (gene_name in c('NES', 'DCX', 'GFAP', 'MAP2', 'TUBB3', 'TH', 'FOXG1', 'TBR1', 'EMX1', 'ACTB')) {
  gene_id <- GetGeneID(gene_name)
  print(PlotExpression(gene_id, EarlySamples, gene_name))
  pvalues <- bind_rows(pvalues, NoteSig(gene_id, gene_name, EarlySamples, LateSamples, All_Samples))
}
kable(pvalues)

```
```{r}
pvalues <- InitializePvalues()
for (gene_name in c('MYOT', 'IGLV4-69', 'NR2F6', 'TOPORS')) {
  gene_id <- GetGeneID(gene_name)
  print(PlotExpression(gene_id, All_Samples, gene_name))
  pvalues <- bind_rows(pvalues, NoteSig(gene_id, gene_name, EarlySamples, LateSamples, All_Samples))
}
kable(pvalues)
```
