
---
title: "Allele-specific Expression"
author: "Heath O'Brien"
output:
  tufte::tufte_html: default
  #tufte::tufte_handout: default
---
<!-- see http://rstudio.github.io/tufte/ for info about tufte -->
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warnings=FALSE, message = FALSE, dev.args = list(bg = 'transparent'))
rm(list=ls())

library(readr)
library(dplyr)
library(DEXSeq)
library(BiocParallel)
```

- 15533_2 (duplicate), 17025 (low read numbers), 17198 (poor mapping quality) and 18121 and 16972 (sample swaps) excluded
- No sexseq-count data for 15468, 15768, 16548, 16640, 16649 and 17081

```{r }
BPPARAM = MulticoreParam(workers=8)
countFiles <- list.files("~/BTSync/FetalRNAseq/Counts/DEXSeq", pattern=".chr.nsort.dex_counts.txt$", full.names=TRUE)
flattenedFile <- "~/BTSync/FetalRNAseq/Reference/genes.gff"
sampleTable <- read_tsv("~/BTSync/FetalRNAseq/Counts/DEXSeq/target.txt")
sample_info <- read_tsv("~/BTSync/FetalRNAseq/Info/sample_info.txt")
sampleTable <- left_join(sampleTable, dplyr::select(sample_info, BrainBankID, Sex, PCW, RIN), by = c("label" = "BrainBankID"))
rownames(sampleTable) <- sampleTable$label
#sampleTable <-sampleTable[1:6,]
sampleTable <- as.data.frame(sampleTable)
dxd = DEXSeqDataSetFromHTSeq(countFiles, #[1:6],
                             sampleData=sampleTable,
                             design= ~ sample + exon + Sex:exon,
                             flattenedfile=flattenedFile
                             )
formulaFullModel    =  ~ sample + exon + centre:exon + PCW:exon + RIN:exon + Sex:exon 
formulaReducedModel    =  ~ sample + exon + centre:exon + PCW:exon + RIN:exon
dxd = estimateSizeFactors( dxd )
dxd = estimateDispersions( dxd, BPPARAM=BPPARAM ) #, formula = formulaFullModel, )
png(filename="~/BTSync/FetalRNAseq/Counts/DEXSeq/DispEsts.png", bg="transparent", width=300, height=300, units="px")
plotDispEsts( dxd )
dev.off()
dxd = testForDEU( dxd, BPPARAM=BPPARAM)
#                  reducedModel = formulaReducedModel,
#                  fullModel = formulaFullModel,
#                  
#)

dxd = estimateExonFoldChanges( dxd, fitExpToVar="Sex", BPPARAM=BPPARAM)
dxr1 = DEXSeqResults( dxd )
write.table(dxr1, file = "~/BTSync/FetalRNAseq/Counts/DEXSeq/results.txt")
table ( dxr1$padj < 0.05 ) # number of significant features (FDR < 0.1)
table ( tapply( dxr1$padj < 0.05, dxr1$groupID, any ) ) # number of genes affected
png(filename="~/BTSync/FetalRNAseq/Counts/DEXSeq/plotMA.png", bg="transparent", width=300, height=300, units="px")
plotMA( dxr1, cex=0.8 )
dev.off()
DEXSeqHTML( dxr1, FDR=0.1, color=c("#FF000080", "#0000FF80"), fitExpToVar="Sex", path="~/BTSync/FetalRNAseq/Counts/DEXSeq", BPPARAM=BPPARAM )
```