---
title: "Data QC"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE)
library(tidyr)
library(ggplot2)
library(dplyr)
library(gridExtra)
library(readr)


rm(list=ls())
```
## Read numbers
- One sample (17025 = 150610_D00125_0239_BC718AANXX_3_IL-TP-013_1/2) has almost no reads.
- One other sample has < 50 million. All others are >60 million with most being > 100 million
- Exeter samples are mcentered around 150 million reads while Edinburgh samples are centered around ca. 120 million
- Once the difference in read length is accounted for, they have similar distributions

```{r}

lengths <- read_delim("/Users/heo3/BTSync/FetalRNAseq/FastQC/seq_lengths.txt", 
                      delim='\t', 
                      col_names=c('file', 'num_reads'), 
                      col_types=cols(file='c', num_reads='i')
                      )
lengths$Million_reads <- lengths$num_reads/1000000
SeqInfo <- read_delim("/Users/heo3/BTSync/FetalRNAseq/LabNotes/sequences.txt", 
                     delim='\t',
                     col_names=c('file', 'read_group', 'centre'),
                     col_types=cols(file='c', read_group='c', centre='c')
                     )

# If I do a full_join here, there are some issues of missing data. I cannot rerun FastQC ATM becuase the data aren't mounted on the compute nodes. I will have to revisit once this is fixed
#full_join(SeqInfo, lengths, by='file') %>% View()

lengths <- inner_join(SeqInfo, lengths, by='file') 
lengths<-mutate(lengths, Billion_bp = ifelse(centre == 'Exeter', 
                                    Million_reads * 100 / 1000, 
                                    ifelse(centre == 'Edinburgh', 
                                           Million_reads * 125 / 1000, 
                                           Million_reads * 75 / 1000
                                           )
                                    )
       )
lengths <- separate(lengths, read_group, c('sample', 'run'), fill='right', remove=FALSE) 
lengths <- group_by(lengths, centre, sample) %>% summarise(Million_reads = sum(Million_reads), Billion_bp = sum(Billion_bp))
layout(matrix(c(1,2), 2, byrow = TRUE))
p1<-ggplot(lengths, aes(x=Million_reads, fill=centre))+geom_histogram(bins=25)+facet_grid(centre ~ .) + theme(legend.position="none")
p2<-ggplot(lengths, aes(x=Billion_bp, fill=centre))+geom_histogram(bins=30)+facet_grid(centre ~ .) + theme(legend.position="none")
grid.arrange(p1, p2, ncol=2)
```

## Fast QC

- All samples pass Basic stats, N content, Per-seq quality and length dist
- All samples fail or warn for per-base sequence content
- All but two fail Kmer content
    - TCTATCGGG/TGGCGCGTGC appears at start of read 1/2 40,000 + times
- Most get a warning for Overrepresented seqs but 6 fail (one pairs, four singletons )
- Only 7 samples pass GC content and only 5 pass Sequence Duplication Levels (about equal FAIL and WARN)
- Two samples from 150521 fail per tile quality (slight overloading?)
- 17921-l1_CGATGT_L006_R2_001 fails Per base quality (this is bacause the first base has low quality).
- 150610_D00125_0239_BC718AANXX_3_IL-TP-013_2.sanfastq looks horible. 150610_D00125_0239_BC718AANXX_3_IL-TP-013_1.sanfastq has fine quality, but A is really over-represented
- Edinburgh sequences have a lot of adaptor sequence

```{r fig.width=8, fig.height=32}

summary_all <- read.delim("/Users/heo3/BTSync/FetalRNAseq/FastQC/summary.txt", header=FALSE, stringsAsFactors=FALSE)
fastQC_summary <- spread(summary_all, V2, V1)
colnames(fastQC_summary) <- c("Sample", "Adapter", "Basic_stats", "Kmers", "Overrepresented", "N_content", "Per_base_content", "Per_base_Q", "Per_seq_GC", "Per_seq_Q", "Per_tile_Q", "Duplication", "Length_dist")
#summary(fastQC_summary[-1]) #this no longer works since I set stringsAsFactors to FALSE
ggplot(summary_all, aes(x=V2, y=V3, fill=V1)) +
  geom_tile() +
  scale_fill_manual(values=c('red', 'green', 'yellow')) +
  theme(axis.text.x= element_text(angle=90, hjust = 1), axis.text.y= element_text(size=8), axis.title = element_blank(), legend.title=element_blank())
```

## Rerun on trimmed sequences

All green in the adaptor column

```{r fig.width=8, fig.height=20}

summary_all <- read.delim("/Users/heo3/BTSync/FetalRNAseq/FastQC/trimmed_summary.txt", header=FALSE, stringsAsFactors=FALSE)
fastQC_summary <- spread(summary_all, V2, V1)
colnames(fastQC_summary) <- c("Sample", "Adapter", "Basic_stats", "Kmers", "Overrepresented", "N_content", "Per_base_content", "Per_base_Q", "Per_seq_GC", "Per_seq_Q", "Per_tile_Q", "Duplication", "Length_dist")
#summary(fastQC_summary[-1]) #this no longer works since I set stringsAsFactors to FALSE
ggplot(summary_all, aes(x=V2, y=V3, fill=V1)) +
  geom_tile() +
  scale_fill_manual(values=c('red', 'green', 'yellow')) +
  theme(axis.text.x= element_text(angle=90, hjust = 1), axis.text.y= element_text(size=8), axis.title = element_blank(), legend.title=element_blank())
```
