---
title: "Female vs Male"
output: html_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(gridExtra)
library(clusterProfiler)
library(GGally)
LabNotes="/Users/heo3/BTSync/FetalRNAseq/LabNotes/"
source(paste0(LabNotes, 'R/AnalyseDE.R'))
source("~/BTSync/FetalRNAseq/LabNotes/R/FormatGGplot.R")
MalevsFemale_complete <- GetCounts()
```

## Histogram of sample size per week/sex
```{r}
target <- GetTarget(12, 20)

target %>% group_by(Sex) %>% summarise(mean(PCW))
png(file="~/BTSync/FetalRNAseq/LabNotes/Results/PCWhist.png", bg="transparent", width=128, height=96, units='mm', res=227)
layout(matrix(c(1,2), 2, byrow = TRUE))
p1 <- ggplot(target, aes(x=PCW, fill=Sex)) +
  geom_bar() +
  facet_grid(Sex ~ .) +
  tufte_theme() +
  scale_y_continuous(breaks=seq(0,100, 2)) +
  scale_x_continuous(breaks=seq(11,19, 1)) +
  tufte_theme() +
  xlab("Post Conception Weeks") +
  theme(axis.text.x=element_text(size=8)) +
  scale_fill_brewer(type = "qual", palette = 6) 

p2 <- mutate(target, age_bin =ifelse(target$PCW == 16, 15, 
                                     ifelse(target$PCW > 16, 16, target$PCW))) %>%
ggplot(aes(x=age_bin, fill=Sex)) +
  geom_bar() +
  facet_grid(Sex ~ .) +
  tufte_theme() +
  scale_y_continuous(breaks=seq(0,100, 2)) +
  scale_x_continuous(breaks=c(11,12,13,14,15,16),
                     labels=c('11','12','13','14','15-16','17-19')) +
  tufte_theme() +
  xlab("Post Conception Weeks") +
  theme(axis.text.x=element_text(size=8)) +
  scale_fill_brewer(type = "qual", palette = 6) 

grid.arrange(p1, p2, ncol=2)
dev.off()
grid.arrange(p1, p2, ncol=2)
```


## R Markdown

- 51 million - 353 million read pairs (14 billion total)
- 28 million - 252 million properly mapped pairs (10.7 billion total)
- 5.3 million - 114 million read pairs included in counts matrix (3.5 billion total)

- only sample with < 10 million read pairs in count matrix (17046) has 78% of reads mapping to rDNA
- two additional samples with <10% of read pairs in count matrix have ~50 or reads mapping to rDNA


```{r }
count_matrix <- MalevsFemale_complete %>%
  dplyr::select(-Id, -starts_with('norm'), -baseMean, -Female, -Male, -FC, -log2FoldChange, -pvalue, -padj, -tagwise.dispersion, -trended.dispersion, -Chr, -ChrType)
Counts <- count_matrix %>% 
  tidyr::gather(Sample, Count) %>% 
  mutate(Sample = as.character(Sample)) %>% 
  group_by(Sample) %>%
  dplyr::summarise(TotalCount=sum(Count))

lengths <- read_delim("/Users/heo3/BTSync/FetalRNAseq/FastQC/seq_lengths.txt", 
                      delim='\t', 
                      col_names=c('read_file', 'num_reads'), 
                      col_types=cols(read_file='c', num_reads='i')
                      )
SeqInfo <- read_delim(paste0(LabNotes, "sequences.txt"), 
                     delim='\t',
                     col_names=c('read_file', 'read_group', 'centre', 'folder'),
                     col_types=cols(read_file='c', read_group='c', centre='c', folder='c')
                     )


lengths <- full_join(SeqInfo, lengths, by='read_file') %>%
  separate(read_group, c('Sample', 'run'), fill='right', remove=FALSE) 

#Sample 18208 was orinigally labelled 18121
lengths[lengths[, 'Sample'] =='18121',]$Sample <- '18208'

counts_files <- read_delim(paste0(LabNotes, "chr_bam_R1.txt"),
"\t", escape_double = FALSE, col_names = c('file_path', 'counts_file', 'read_path', 'read_file'),
trim_ws = TRUE)

mapping_stats <- read_delim(paste0(LabNotes, "Results/Mapping_stats.txt"),
" ", escape_double = FALSE, col_names = TRUE,
trim_ws = TRUE)

combined <- lengths %>% 
  filter(grepl('_1$', read_file) | grepl('_R1_', read_file)) %>%
  full_join(counts_files)

stats <- combined %>% dplyr::select(Sample, file_path) %>%
  right_join(mapping_stats) %>% 
  distinct() %>%
  dplyr::select(-in_stats, -ex_stats, -file_path) %>%
  group_by(Sample) %>%
  summarise_each(funs(sum))

total_reads <- combined %>%
  group_by(Sample) %>%
  dplyr::summarise(TotalFragments=sum(as.numeric(num_reads)))

final_stats <- total_reads %>%
  full_join(stats) %>%
  right_join(Counts) %>% 
  mutate(MappedFragments = Paired/2, rDNA_Fragments= floor(rDNA/2)) %>%
  dplyr::select(Sample, TotalFragments, rDNA_Fragments, MappedFragments, TotalCount) %>%
  mutate(perc_rDNA = rDNA_Fragments/TotalFragments*100,
         perc_mapped = MappedFragments/TotalFragments*100,
         perc_counted = TotalCount/TotalFragments*100)

final_stats %>% arrange(TotalCount)
final_stats %>%
  dplyr::select(TotalFragments, MappedFragments, TotalCount) %>%
  summarise(TotalFragments=sum(as.numeric(TotalFragments))/1000000000,
            MappedFragments=sum(as.numeric(MappedFragments)/1000000000),
            TotalCount=sum(as.numeric(TotalCount)/1000000000)
  )

```

- In full 105 sample dataset, 47524 of 66023 ENSEMBL features have at least one read mapping
- 17070 features with >=1 Counts Per Million in at least as many samples as the min of male/famale were included in the differential expression analysis
```{r}
Fold_changes <- GetFoldChanges()
CPM1 <- Fold_changes %>%
  filter(! is.na(log2FoldChange)) %>%
  group_by(week) %>%
  dplyr::summarise(n=n()) #%>%
  #mutate(week = ifelse(week == 15.5, 15, ifelse(week==18,16,week)))

```

```{r}
feature_stats <- data.frame(Samples=c(ncol(count_matrix)), TotalFeatures=c(nrow(count_matrix)))
feature_stats <- count_matrix %>%
  mutate(sum=rowSums(.)) %>%
  filter(sum > 0) %>%
  dplyr::summarise(NonZeroFeatures=n()) %>%
  bind_cols(feature_stats)
feature_stats <- MalevsFemale_complete %>%
  filter(! is.na(FC)) %>%
  dplyr::summarise(IncludedFeatures=n()) %>%
  bind_cols(feature_stats)
  
dplyr::select(feature_stats, Samples, TotalFeatures, NonZeroFeatures, IncludedFeatures)

p1 <- count_matrix %>%
  gather(Sample, Count) %>%
  mutate(Sample = as.character(Sample)) %>%
  filter(Count > 0) %>%
  group_by(Sample) %>%
  dplyr::summarise(n=n()) %>%
  full_join(dplyr::select(target, label, PCW, Sex), by=c('Sample' = 'label')) %>%
  ggplot(aes(x=PCW, y=n)) +
    geom_bar(aes(x=week, y=n), stat='identity', data=CPM1) +
    geom_jitter(aes(colour=Sex), height=0, width=.25) +
    tufte_theme() +
    scale_y_continuous(limits=c(0,66000), breaks=seq(0,60000,10000)) +
    scale_x_continuous(breaks=seq(12, 19, 1)) +
    xlab("Post Conception Weeks") +
    ylab("Ensembl transcripts (66023 total)") +
    theme(axis.text.x=element_text(size=8)) +
    scale_colour_brewer(type = "qual", palette = 6) +
    theme(legend.position="right")
png(file="~/BTSync/FetalRNAseq/LabNotes/Results/NumTranscripts.png", bg="transparent", width=128, height=96, units='mm', res=227)
p1
dev.off()  
p1

```

- Total of 452 DE (FDR<0.1) genes when all 105 samples analysed together
- 261 are higher in males and 191 are higher in Females
- Of these, 92 are also DE in at least one of the samples subsetted by age
- 41 are on chrX and 20 are on chrY


```{r}
#Genes that are DE in combined analysis
all_DE <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_HDBR_excl_15641_18432_16491_PCW_FDR_0.1_edgeR/tables/FemaleUp.txt",
"\t", escape_double = FALSE, trim_ws = TRUE) %>%
  bind_rows(read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_12_20_HDBR_excl_15641_18432_16491_PCW_FDR_0.1_edgeR/tables/MaleUp.txt",
"\t", escape_double = FALSE, trim_ws = TRUE)) %>%
  mutate(Id = gsub('\\..*', '', Id))
all_DE_by_week <- all_DE %>% 
  dplyr::select(-Male, -Female, -FC, -log2FoldChange) %>%
  left_join(Fold_changes, by='Id') %>%
  mutate(gene_name = GeneID, ChrType=ifelse(Chr=='chrX', 'chrX', ifelse(Chr=='chrY', 'chrY', 'autosome')))

all_DE_by_week_matrix <- all_DE_by_week %>% 
  dplyr::select(Id, Chr, GeneID, sig, week) %>%
  spread(week, sig) %>% 
  mutate(total = rowSums(.[4:8]))

all_DE_by_week_matrix <- all_DE_by_week_matrix %>%
  full_join(dplyr::select(all_DE, Id, log2FoldChange))

all_DE_by_week_matrix %>%
  arrange(desc(total)) %>% 
  dplyr::select(-total)

dplyr::summarise(all_DE_by_week_matrix, TotalDE = n()) %>%
  bind_cols(filter(all_DE_by_week_matrix, log2FoldChange > 0) %>%
  dplyr::summarise(MaleUp = n())) %>%
  bind_cols(filter(all_DE_by_week_matrix, log2FoldChange < 0) %>%
  dplyr::summarise(FemaleUp = n())) %>%
  bind_cols(filter(all_DE_by_week_matrix, Chr == 'chrY') %>% 
  dplyr::summarise(ChrY = n())) %>%
  bind_cols(filter(all_DE_by_week_matrix, Chr == 'chrX') %>% 
  dplyr::summarise(ChrX = n())) %>%
  bind_cols(filter(all_DE_by_week_matrix, total > 0) %>%
  dplyr::summarise(DE_by_week = n()))


PlotFC_by_ChrType(all_DE_by_week, 'All DE')
all_DE_by_week %>% filter(ChrType != 'chrY' & gene_name != 'XIST') %>%
  PlotFC_by_ChrType("Non Y/XIST DE")
```

```{r}
p1 <- all_DE %>%
    mutate(ChrType=ifelse(Chr=='chrX', 'chrX', ifelse(Chr=='chrY', 'chrY', 'autosome')), Sex = ifelse(log2FoldChange < 0, 'F', 'M')) %>%
  ggplot(aes(x=Sex, fill=ChrType)) + geom_bar(position='stack') + 
    tufte_theme() +
    scale_y_continuous(breaks=seq(100,1000,100)) +
    scale_fill_manual(values=brewer.pal(6, "Set1")[c(1,2,4)]) +
    theme(legend.position='right') +
    theme(strip.switch.pad.grid = unit(10, 'cm')) +
    ylab('Number DE genes (FDR<0.1)') +
    xlab('Fetal Sex') +
    ggtitle("All Ages (12-19 PCW)")
  
png(file="~/BTSync/FetalRNAseq/LabNotes/Results/AllSigBar.png", bg="transparent", width=128, height=96, units='mm', res=227)
p1
dev.off()
p1

p2 <- Fold_changes %>% filter(sig == 1) %>%
  mutate(week = paste(ifelse(week > 16, 
                      '15-16',
                      ifelse(week >= 15, 
                             '17-19', 
                             week)), 'PCW'),
         Sex = ifelse(log2FoldChange < 0, 'F', 'M')) %>% 
  left_join(dplyr::select(MalevsFemale_complete, Id, ChrType)) %>%
  ggplot(aes(x=Sex, fill=ChrType)) + geom_bar(position='stack') + 
  facet_grid(. ~ week) +
    tufte_theme() +
    scale_y_continuous(breaks=seq(100,1000,100)) +
    scale_fill_manual(values=brewer.pal(6, "Set1")[c(1,2,4)]) +
    theme(legend.position='right') +
    theme(strip.switch.pad.grid = unit(10, 'cm')) +
    ylab('Number DE genes (FDR<0.1)') +
    xlab('Fetal Sex')
png(file="~/BTSync/FetalRNAseq/LabNotes/Results/ByWeekSigBar.png", bg="transparent", width=128, height=96, units='mm', res=227)
p2
dev.off()
p2

```

```{r}
p1 <- MalevsFemale_complete %>% 
  dplyr::select(Id, ChrType) %>%
  inner_join(Fold_changes, by='Id') %>%
  PlotFC_by_ChrType("All genes CMP >= 1 in 45 or more samples")
png(file="~/BTSync/FetalRNAseq/LabNotes/Results/AllGenesFC.png", bg="transparent", width=128, height=96, units='mm', res=227)
p1
dev.off()
p1
p2<-MalevsFemale_complete %>% 
  dplyr::select(Id, ChrType) %>%
  filter(ChrType != 'chrY' & Id != 'ENSG00000229807') %>% 
  inner_join(Fold_changes, by='Id') %>%
  PlotFC_by_ChrType("All Non Y/XIST genes")

png(file="~/BTSync/FetalRNAseq/LabNotes/Results/AllNonYGenesFC.png", bg="transparent", width=128, height=96, units='mm', res=227)
p2
dev.off()
p2
```

```{r}
gene_sets <- GetGeneSets()
gene_sets <- dplyr::select(MalevsFemale_complete, ENSEMBL = Id, ChrType) %>%
  right_join(gene_sets)

Neuronal <- gene_sets %>% 
  filter(set=='Neuronal' & gene_name != 'XIST') %>%
  left_join(Fold_changes, by=c('ENSEMBL' = 'Id')) %>%
  mutate(Id = ENSEMBL)
NeuronalFC <- PlotFC(Neuronal, 'Neuronal Genes')
png(file="~/BTSync/FetalRNAseq/LabNotes/Results/Neuronal.png", bg="transparent", width=128, height=96, units='mm', res=227)
NeuronalFC
dev.off()
NeuronalFC

NPC <- gene_sets %>% 
  filter(set=='NPC' & gene_name != 'RPS4Y1') %>%
  left_join(Fold_changes, by=c('ENSEMBL' = 'Id')) %>%
  mutate(Id = ENSEMBL)
NPCFC <- PlotFC(NPC, 'Neural Progenitor Cell Genes')
png(file="~/BTSync/FetalRNAseq/LabNotes/Results/NPC.png", bg="transparent", width=128, height=96, units='mm', res=227)
NPCFC
dev.off()
NPCFC

RG <- gene_sets %>% 
  filter(set=='RadialGlia' & gene_name != 'RPS4Y1') %>%
  left_join(Fold_changes, by=c('ENSEMBL' = 'Id')) %>%
  mutate(Id = ENSEMBL)
RGFC<-PlotFC(RG, 'Radial Glia Genes')
png(file="~/BTSync/FetalRNAseq/LabNotes/Results/RadialGlia.png", bg="transparent", width=128, height=96, units='mm', res=227)
RGFC
dev.off()
RGFC

AN <- gene_sets %>% 
  filter(set=='cluster7' & gene_name != 'RPS4Y1') %>%
  left_join(Fold_changes, by=c('ENSEMBL' = 'Id')) %>%
  mutate(Id = ENSEMBL)
ANFC<-PlotFC(AN, 'Top 21 Adult Neuron Genes (cluster 7)')
png(file="~/BTSync/FetalRNAseq/LabNotes/Results/AdultNeuron.png", bg="transparent", width=128, height=96, units='mm', res=227)
ANFC
dev.off()
ANFC

FR <- gene_sets %>% 
  filter(set=='cluster9' & gene_name != 'RPS4Y1') %>%
  left_join(Fold_changes, by=c('ENSEMBL' = 'Id')) %>%
  mutate(Id = ENSEMBL)
FRFC<-PlotFC(FR, 'Top 21 Replicating Fetal Genes (cluster 9)')
png(file="~/BTSync/FetalRNAseq/LabNotes/Results/FetalReplicating.png", bg="transparent", width=128, height=96, units='mm', res=227)
FRFC
dev.off()
FRFC
```

```{r}

disease2gene <- gene_sets[, c("set", "gene_name")] 

FemaleDEG <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBR_FDR_0.1_edgeR/tables/FemaleUp.txt",
"\t", escape_double = FALSE, trim_ws = TRUE)

FemaleEnrichment <- summary(enricher(FemaleDEG$GeneID, TERM2GENE=disease2gene))
FemaleEnrichment$Direction <- 'FemaleBiased'

MaleDEG <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBR_FDR_0.1_edgeR/tables/MaleUp.txt",
"\t", escape_double = FALSE, trim_ws = TRUE)

MaleEnrichment <- summary(enricher(MaleDEG$GeneID, TERM2GENE=disease2gene))
MaleEnrichment$Direction <- 'MaleBiased'
Enrichment <- bind_rows(FemaleEnrichment, MaleEnrichment) %>% 
  separate(GeneRatio, c('x1', 'y1'), sep='/', remove=FALSE) %>% 
  separate(BgRatio, c('x2', 'y2'), sep='/', remove=FALSE) %>%
  mutate(Enrichment=(as.numeric(x1)/as.numeric(y1))/(as.numeric(x2)/as.numeric(y2))) %>%
  dplyr::select(Direction, ID, GeneRatio, BgRatio, Enrichment, pvalue, p.adjust, qvalue, geneID)
write_tsv(Enrichment, "~/BTSync/FetalRNAseq/LabNotes/Results/Enrichment.txt")
Enrichment
```
```{r}
dplyr::select(MaleEnrichment@result, ID, Count, geneID)
PlotExpression('MCM2', MalevsFemale_complete, '17921')
PlotExpression('NEUROD4', MalevsFemale_complete, '17921')
PlotExpression('ESPL1', MalevsFemale_complete, '17921')
PlotExpression('VIM', MalevsFemale_complete, '17921')
PlotExpression('AASS', MalevsFemale_complete, '17921')
PlotExpression('FAM64A', MalevsFemale_complete, '17921')
PlotExpression('GLI2', MalevsFemale_complete, '17921')
PlotExpression('TFAP2C', MalevsFemale_complete, '17921')
PlotExpression('LIPG', MalevsFemale_complete, '17921')
```
```{r}
dplyr::select(FemaleEnrichment@result, ID, Count, geneID)
PlotExpression('ATP1B1', MalevsFemale_complete, '17921')
PlotExpression('NAP1L5', MalevsFemale_complete, '17921')
PlotExpression('LGI1', MalevsFemale_complete, '17921')
PlotExpression('MAGEE1', MalevsFemale_complete, '17921')
PlotExpression('NAP1L3', MalevsFemale_complete, '17921')
PlotExpression('CCK', MalevsFemale_complete, '17921')
PlotExpression('RYR2', MalevsFemale_complete, '17921')
PlotExpression('FGF12', MalevsFemale_complete, '17921')
PlotExpression('SCG2', MalevsFemale_complete, '17921')
```
```{r}
PlotExpression('NES', MalevsFemale_complete, '17921')
PlotExpression('DCX', MalevsFemale_complete, '17921')
PlotExpression('GRIN2A', MalevsFemale_complete, '17921')
PlotExpression('HTR1B', MalevsFemale_complete, '17921')
PlotExpression('GRIN3A', MalevsFemale_complete, '17921')
PlotExpression('HTR2A', MalevsFemale_complete, '17921')
PlotExpression('GABRG2', MalevsFemale_complete, '17921')
```

- Very low/negative fold-change correlation between weeks (when Y/XIST excluded)
- For some reason, week 17-19 is correlated with everything
- When week 14 is split into two samples, the second set is highly correlated with the full set (r=0.904), but the correlation with the first set is much weeker (r=0.675), and there are some serious outliers (one sample with log2FC=-10 for this analysis vs close to 0 for all others)


```{r}
Excluded <- MalevsFemale_complete %>% filter(ChrType == 'chrY' | Id == 'ENSG00000229807') %>%
  dplyr::select(Id)
W14_DE <-
FC_ByWeek <- Fold_changes %>%
  dplyr::select(-sig) %>%
  spread(week, log2FoldChange) %>% 
  dplyr::select(Id, 'w12'=`12`, 'w13'=`13`, 'w14'=`14`, 'w15_16'=`15.5`, 'w17_19'=`18`) %>%
  full_join(
    read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBR_excl_19043_17921_19031_17221_18015_17053_17071_18529_18856_FDR_0.1_edgeR/tables/MalevsFemale.complete.txt",
"\t", escape_double = FALSE, trim_ws = TRUE) %>%
  dplyr::select(Id, w14_1= log2FoldChange) %>%
  separate(Id, c('Id'), sep = "[.]", extra='drop')
  ) %>%
  full_join(
    read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBR_excl_17812_18282_18349_18355_18687_16428_16488_16640_16929_FDR_0.1_edgeR/tables/MalevsFemale.complete.txt",
"\t", escape_double = FALSE, trim_ws = TRUE) %>% 
  dplyr::select(Id, w14_2=log2FoldChange) %>%
  separate(Id, c('Id'), sep = "[.]", extra='drop')
  ) %>%
  full_join(
    read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBR_excl_17812_18282_18349_17221_18015_16428_16488_18529_18856_FDR_0.1_edgeR/tables/MalevsFemale.complete.txt",
"\t", escape_double = FALSE, trim_ws = TRUE) %>% 
  dplyr::select(Id, w14_3=log2FoldChange) %>%
  separate(Id, c('Id'), sep = "[.]", extra='drop')
  ) %>%
  full_join(
    read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBR_excl_19043_17921_19031_18355_18687_17053_17071_16640_16929_FDR_0.1_edgeR/tables/MalevsFemale.complete.txt",
"\t", escape_double = FALSE, trim_ws = TRUE) %>% 
  dplyr::select(Id, w14_4=log2FoldChange) %>%
  separate(Id, c('Id'), sep = "[.]", extra='drop')
  ) %>%
  anti_join(Excluded) 
dplyr::select(FC_ByWeek, -Id, -starts_with('w14_')) %>%
  ggpairs()
dplyr::select(FC_ByWeek, starts_with('w14')) %>%
  ggpairs()
filter(Fold_changes, week==14 & sig==1) %>% 
  inner_join(FC_ByWeek) %>% 
  dplyr::select(starts_with('w14')) %>% 
  ggpairs()
```

- For some reason, the second set is getting some genes with relatively small differences between Male and Female counts, but very large fold changes
- The fold changes are more complex estimates, so simply dividing Male by Female doesn't give the exact same numeber (and in fact I had to add 1 to deal with division by zero errors)
- However, in the case of the second set, fold changes calculated this way correlate highely with the original estimates (r=0.903 with no major outliers)
- The correlation is much lower in the case of the first set (r=.591, with a point at 0,-10)
- In the case of the second set, the original FC values correlate better with the full set than the recalculated ones (r=0.904 vs. r=0.847), while in the case of set 1, the correlation is lower (r=0.675 vs. r=0.733)
- It seems pretty clear that something went horribly wrong in the first run 

```{r}
recalculated <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBR_excl_19043_17921_19031_17221_18015_17053_17071_18529_18856_FDR_0.1_edgeR/tables/MalevsFemale.complete.txt",
"\t", escape_double = FALSE, trim_ws = TRUE) %>% 
  separate(Id, c('Id'), sep = "[.]", extra='drop') %>%
  mutate(w14_1_2 = log2((Male+1)/(Female+1))) %>%
  dplyr::select(Id, w14_1_2) %>%
  inner_join(
    read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBR_excl_17812_18282_18349_18355_18687_16428_16488_16640_16929_FDR_0.1_edgeR/tables/MalevsFemale.complete.txt",
"\t", escape_double = FALSE, trim_ws = TRUE) %>% 
  separate(Id, c('Id'), sep = "[.]", extra='drop') %>%
  mutate(w14_2_2 = log2((Male+1)/(Female+1))) %>%
  dplyr::select(Id, w14_2_2)
  ) %>%
  inner_join(FC_ByWeek)

recalculated %>% dplyr::select(w14, w14_1, w14_1_2, w14_2, w14_2_2) %>%
    ggpairs()

```

- The correlation with the replication set is ...not good
- It's similar to the between-weeks values

```{r}
HDBRexpression <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBRexpression_FDR_0.1_edgeR/tables/MalevsFemale.complete.txt",
"\t", escape_double = FALSE, trim_ws = TRUE) %>% dplyr::select(Id, w14_rep=log2FoldChange) %>% separate(Id, c('Id'), sep = "[.]", extra='drop') %>%
  inner_join(recalculated) 

HDBRexpression %>% dplyr::select(w14, w14_1_2, w14_2, w14_rep) %>%
    ggpairs()

```


```{r}
DEseq_FC <- read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBR_FDR_0.1_DEseq/tables/MalevsFemale.complete.txt",
"\t", escape_double = FALSE, trim_ws = TRUE) %>%
  dplyr::select(Id, w14= log2FoldChange) %>%
  separate(Id, c('Id'), sep = "[.]", extra='drop') %>%
  full_join(
    read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBR_excl_19043_17921_19031_17221_18015_17053_17071_18529_18856_FDR_0.1_DEseq/tables/MalevsFemale.complete.txt",
"\t", escape_double = FALSE, trim_ws = TRUE) %>%
  dplyr::select(Id, w14_1= log2FoldChange) %>%
  separate(Id, c('Id'), sep = "[.]", extra='drop')
  ) %>%
  full_join(
    read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBR_excl_18282_18349_18355_18687_16428_16488_16640_16929_FDR_0.1_DEseq/tables/MalevsFemale.complete.txt",
"\t", escape_double = FALSE, trim_ws = TRUE) %>% 
  dplyr::select(Id, w14_2=log2FoldChange) %>%
  separate(Id, c('Id'), sep = "[.]", extra='drop')
  ) %>%
    full_join(
    read_delim("~/BTSync/FetalRNAseq/Counts/MvsF_14_HDBR_excl_17921_FDR_0.1_DEseq/tables/MalevsFemale.complete.txt",
"\t", escape_double = FALSE, trim_ws = TRUE) %>% 
  dplyr::select(Id, w14_3=log2FoldChange) %>%
  separate(Id, c('Id'), sep = "[.]", extra='drop')
  ) %>%
  anti_join(Excluded) 
dplyr::select(DEseq_FC, -Id) %>%
  ggpairs()
```

